---
title: "**Multigenerational adversity impacts on gut microbiome composition and socioemotional functioning in early childhood**: Main Analyses"
subtitle: "Updated based on PNAS Reviewer Comments"
author: "Fran Querdasi"
date: "7/20/2022"
output: 
  html_document:
    toc: true
    toc_float: true
    theme: united
---

```{r setup-chunk, include=FALSE}
knitr::opts_chunk$set(cache = TRUE, #set cache to be saved so that knitting is faster once it has been done once
                      dev="svg", #set graphs to be saved as. svg (vector-based format rather than raster based formats like.jpg)
                      dpi=1000) #set resolution to be 1000 dpi 
#all graphs will be saved in "files" folder in same directory as script
```

# Setup
## Set main seed 
So it's same for all instances
```{r}
main.seed = 1028
```

## Load libraries  
Using`r R.version$version.string` (4.2.3)
```{r load libraries, results='hide'}
suppressPackageStartupMessages({
  library(vegan)
  library(tidyverse)
  library(stats)
  library(phyloseq)
  library(mefa)
  library(qiime2R)
  library(microbiome)
  library(fantaxtic)
  library(pals)
  library(stringr)
  library(car)
  library(rstatix)
  library(ggpubr)
  library(patchwork) #for combining plots
  library(corrplot)
  library(readxl)
  library(psych)
  library(sjPlot)
  library(RVAideMemoire)
  library(purrr)
  library(olsrr)
  library(ggformula)
  library(reshape2)
  library(mosaic)
  library(micropower)
  library(kableExtra)
  library(multiplex) #for writing mplus .dat files
  library(gridExtra) #for scatterplots
  library(summarytools) #for chi-square test and tables
  source("../helper_functions/omega_sq_adonis.R") #load function to calculate omega squared for adonis objects
  library(Maaslin2)
  library(ppcor) #for eta-squared calculations 
})
```

## Load data
**Note:** Filepaths not shown
```{r load data, echo=FALSE}
#general
general_mb_path <- "../../data/microbiome/M24_M48/"

#for phyloseq
otu_table_path <- str_c(general_mb_path, "input_files/M2448_feature-table.qza")
tree_path <- str_c(general_mb_path, "input_files/M2448_rep_otu-tree.qza")
taxonomy_path <- str_c(general_mb_path, "input_files/M2448_taxonomy.qza")
metadata_path <- str_c(general_mb_path, "input_files/M24_metadata_typeheader_new_rev.tsv")

#for alpha diversity -- stress
m24_stress_data <- "../../data/analysis_datasets/fran/gusto_M24cs_stress_data_ctqrev.csv"
m48_stress_data <- "../../data/analysis_datasets/fran/gusto_M48_stress_data.csv"
leq_resp_file <- "../../data/stress/GUSTO_Y04_LEQ_clean_20211203.csv"

#covariates
sex_file <- "../../data/general_covariates/infant_sex.xlsx"
bf_file <- "../../data/mb_covariates/GUSTO_breastfeeding_clean_20210623.csv"
delivery_file <- "../../data/mb_covariates/GUSTO_deliverymode_cleaned_20210623.csv"
income_file <- "../../data/general_covariates/GUSTO_income_clean_20210702.csv"
biotics_file <- "../../data/mb_covariates/GUSTO_M24_biotics_clean_20220218.csv"
gestage_file <- "../../data/general_covariates/GA_weight_Ages_at_measures_FormA515_20210719.xlsx"
age_file <- "../../data/general_covariates/GUSTO_age_stoolproxy_20220615.csv"
diet_file <- "../../data/mb_covariates/diet_covariates.csv"

#for alpha diversity -- diversity metrics
shannon_file <- str_c(general_mb_path, "diversity_tsvs/M2448_shannon.tsv")
faith_file <- str_c(general_mb_path, "diversity_tsvs/M2448_faith.tsv")
evenness_file <- str_c(general_mb_path, "diversity_tsvs/M2448_evenness.tsv")
richness_file <- str_c(general_mb_path, "diversity_tsvs/M2448_richness.tsv")

#health symptom files
beh4_file <- "../../data/child_beh/GUSTO_CBCL_scores_Y04_20220516.csv"
beh2_file <- "../../data/child_beh/GUSTO_CBCL_scores_M24_20220516.csv"

#for family and genus levels differential abundance
family_taxonomy_file <- str_c(general_mb_path, "input_files/family_level/family-table.csv")
genus_taxonomy_file <- str_c(general_mb_path, "input_files/genus_level/genus-table.csv")
```

# Data Wrangling
## Create phyloseq object
```{r}
physeq_M2448 <- qza_to_phyloseq(
  features=otu_table_path,
  tree=tree_path,
  taxonomy=taxonomy_path,
  metadata=metadata_path
)

#create a phyloseq object containing only 24M samples
physeq_M24 <- subset_samples(physeq_M2448, Timepoint_months=="24")
```

## Wrangle data for alpha diversity analyses
```{r, results='hide', message=FALSE}
#read in continuous stress data
M24_stress <-
  read_csv(m24_stress_data)

M48_stress <-
  read_csv(m48_stress_data)

leq_resp <-
  read_csv(leq_resp_file) %>% 
  dplyr::select(
    subjid,
    person_filling_leq = person_filling_Y04
  )

#bind these datasets together
stress_data <-
  M24_stress %>% 
  bind_rows(M48_stress) %>% 
  left_join(leq_resp, by = "subjid")

#read in alpha diversity data from qiime2
shannon <-
  read_tsv(shannon_file) %>% 
  dplyr::rename('#SampleID' = ...1)

faith <-
  read_tsv(faith_file) %>% 
  dplyr::rename('#SampleID' = ...1)

evenness <-
  read_tsv(evenness_file) %>% 
  dplyr::rename('#SampleID' = ...1)

richness <-
  read_tsv(richness_file) %>% 
  dplyr::rename('#SampleID' = ...1)

#create dataset with alpha diversity metrics & stress
alpha_diversity_stress <-
  shannon %>% 
  left_join(faith, by = "#SampleID") %>% 
  left_join(evenness, by = "#SampleID") %>% 
  left_join(richness, by = "#SampleID") %>% 
  left_join(stress_data, by = "#SampleID")

#read in child sex data
sex_data <- readxl::read_xlsx(sex_file) %>% 
  dplyr::rename(subjid=SubjectID)

#read in other covariates
bf_data <- read_csv(bf_file) %>% 
  full_join(read_csv(delivery_file), by = "subjid") %>% 
  full_join(read_csv(income_file), by = "subjid") %>% 
  full_join(read_csv(age_file), by = "subjid")

gest_age <- readxl::read_xlsx(gestage_file) %>% 
  dplyr::select(
    subjid = SubjectID,
    GA
  )

biotics <- read_csv(biotics_file)

#read in sequencing batch
meta <- read_tsv(metadata_path)
meta <- meta[-c(1), ] #remove #q2:types header
meta <- 
  meta %>% 
  dplyr::select(
    '#SampleID', 
    Sequencing_batch
    )

#bind covariate data to rest
alpha_diversity_stress <-
  alpha_diversity_stress %>% 
  left_join(sex_data, by = "subjid") %>% 
  left_join(bf_data, by = "subjid") %>% 
  left_join(meta, by = "#SampleID") %>% 
  left_join(biotics, by = "subjid") %>% 
  left_join(gest_age, by = "subjid") %>% 
  left_join(read_csv(diet_file), by = "subjid") %>% 
  dplyr::rename(
    MUFA_. = `MUFA_%`,
    PUFA_. = `PUFA_%`,
    SatFat_. = `SatFat_%` 
  )

#read in cbcl (child SEF)
child_beh4 <- read_csv(beh4_file) %>% 
  dplyr::select(
    subjid,
    age_y4_cbcl = age_y4,
    cbcl_respondent_y4,
    cbcl_gi_y4,
    contains("tot_")
  ) 

child_beh2 <-
  read_csv(beh2_file) %>% 
  dplyr::select(
    subjid,
    age_m24_cbcl = age_m24,
    cbcl_respondent_m24,
    cbcl_gi_m24,
    contains("tot_")
  )

#bind cbcl data to rest
alpha_diversity_stress <-
  alpha_diversity_stress %>% 
  left_join(child_beh4, by = "subjid") %>% 
  left_join(child_beh2, by = "subjid")

#create categorical postnatal stress var because LEQ has too few scale points
alpha_diversity_stress <- 
  alpha_diversity_stress %>% 
  mutate(
    postnatal_stress = case_when(
      timepoint == 48 & child_neg_events_sum > 0 ~ 1,
      timepoint == 48 & child_neg_events_sum == 0 ~ 0,
      timepoint == 24 & leq_b2_events_total_Y04 > 0 ~ 1,
      timepoint == 24 & leq_b2_events_total_Y04 == 0 ~ 0
    )
  ) %>% 
  mutate(postnatal_stress = as.factor(postnatal_stress))

#remove all unused variables
alpha_diversity_stress <- alpha_diversity_stress %>% 
  dplyr::select(
    -contains("neg"),
    -contains("leq"),
    -maltreated_rev,
    -contains("days"),
    -c_age_years_stoolproxy_y4
  )

#create dataset of cases that have usable microbiome data and reported on at least 1 timepoint of adversity exposure to be used in analyses (N=450)
alpha_diversity_stress_usable <-
  alpha_diversity_stress %>% 
  filter(timepoint==24 & !(is.na(STAI_prorated_state_pw26) & is.na(ctq_total_score_25it) & is.na(postnatal_stress)))

#create categorical prenatal adversity variable
alpha_diversity_stress_usable <- alpha_diversity_stress_usable %>% 
  mutate(stai_above_clin_cutoff = ifelse(STAI_prorated_state_pw26 > 40, 1, 0))

#create standardized cbcl scores with mean=100, sd=15 within each sex (scores are named cbcl{subscale}tot_{timepoint}_std) and then remove unstandardized variables
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  group_by(sex_binary) %>% 
  dplyr::mutate(across(c(contains("tot_"), contains("gi")), .fns = list(std = ~as.vector(scale(.))*15 + 100), .names="{col}_{fn}")) %>% 
  ungroup() %>% 
  dplyr::select(
    -(contains("tot_") & !contains("std")),
    -(contains("gi") & !contains("std"))
  )

```
Note on decision to standardize cbcl raw scores within each sex: ASEBA Y1.5-5 manual recommends using raw scores but standardizing by within your sample by sex (pg 122). Their suggested mean is 100, sd is 15. 

## Create cumulative adversity variables
```{r, warning=FALSE}
#create sum of timepoints with "substantial" adversity exposure (range = 0 (no timempoints with substantial adversity exposure) to 3 (substantial adversity exposure at all timepoints))
alpha_diversity_stress_usable <- alpha_diversity_stress_usable %>% 
  dplyr::mutate(
    sum_advrsty = as.numeric(as.character(maltreated_mod_rev)) + stai_above_clin_cutoff + as.numeric(as.character(postnatal_stress))
  )

#determine how many dyads have 0, 1, 2, and 3 timepoints of exposure
alpha_diversity_stress_usable %>% ggplot(aes(x=sum_advrsty, fill=as.factor(sum_advrsty))) +
  geom_histogram(binwidth = 1) +
  stat_bin(aes(y=..count.., label=ifelse(..count..==0,"",..count..)), geom="text", vjust=-.5) +
  scale_fill_discrete("Timepoints with Adversity Exposure") +
  xlab("Number of Timepoints with Adverity Exposure") +
  ylab("Number of Participants") 

#create the collapsed grouping of 3 cumulative adversity levels (because very few dyads had sum_advrsty=3)
alpha_diversity_stress_usable <- alpha_diversity_stress_usable %>% 
  mutate(
    cumulative_adv = case_when(
      sum_advrsty == 0 ~ 0,
      sum_advrsty == 1 ~ 1,
      sum_advrsty == 2 | sum_advrsty == 3 ~ 2
    )
  ) 

#convert categorical binary & multicategorical variables to factor type (for analyses that require it later)
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  dplyr::mutate(across(c(cumulative_adv, sex_binary, deliv_mode, postnatal_stress, Sequencing_batch, maltreated_mod_rev, antibio_M24, probiotics_M24), ~as.factor(.x)))

#write alpha diversity analyses dataset to file
write_csv(alpha_diversity_stress_usable, "../../data/analysis_datasets/fran/alpha_div_stress_usable_ctqrev.csv")
#also write these data to form B data folder for submission with GUSTO form B application
write_csv(alpha_diversity_stress_usable, "../../manuscripts/fran_adversity_microbiome/Form_B_submission/data_updated/alpha_diversity_and_metadata_R1.csv")
```

## Check whether there are differences in selected socioemotional functioning domains by child sex, family income, and ethnicity 
### Socioemotional Functioning at Age 2 Years
```{r eval=FALSE}
#get relevant variables
cbcl_covs <- alpha_diversity_stress_usable %>% 
  dplyr::select(
    monthly_income_per_member,
    GA,
    cbclintprobtot_m24,
    cbcltotprobtot_m24,
    cbcldevprobtot_m24,
    age_m24_cbcl
  )

#correlations with cbcl and continuous covs
corPlot(cbcl_covs,
        numbers = TRUE,
        stars = TRUE,
        show.legend = TRUE,
        cex.axis = 0.8,
        cex = 0.5) 
#no associations between monthly income and cbcl outcome vars

#associations with categorical covs
t.test(cbclintprobtot_m24~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(cbcltotprobtot_m24~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(cbcldevprobtot_m24~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns

summary(aov(cbclintprobtot_m24~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbcltotprobtot_m24~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbcldevprobtot_m24~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
```
No domains differ

### Socioemotional Functioning at Age 4 Years
```{r eval=FALSE}
#get relevant variables
cbcl_covs_y4 <- alpha_diversity_stress_usable %>% 
  dplyr::select(
    monthly_income_per_member,
    GA,
    cbclintprobtot_y4,
    cbcltotprobtot_y4,
    cbclextprobtot_y4,
    cbclattprobtot_y4,
    cbclsleepprobtot_y4,
    cbcldevprobtot_y4,
    cbcladhdprobtot_y4,
    age_y4_cbcl
  )

#correlations with cbcl domains and continuous covariates
corPlot(cbcl_covs_y4,
        numbers = TRUE,
        stars = TRUE,
        show.legend = TRUE,
        cex.axis = 0.8,
        cex = 0.5) 
#no differences

#associations with categorical covs
t.test(cbclintprobtot_y4~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(cbcltotprobtot_y4~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(cbclextprobtot_y4~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #sig, males higher than females
t.test(cbclattprobtot_y4~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #sig, males higher than females
t.test(cbclsleepprobtot_y4~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(cbcldevprobtot_y4~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(cbcladhdprobtot_y4~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns


summary(aov(cbclintprobtot_y4~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbcltotprobtot_y4~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbclextprobtot_y4~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbclattprobtot_y4~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbclsleepprobtot_y4~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbcldevprobtot_y4~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbcladhdprobtot_y4~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns

```
Externalizing and attention problems at year 4 differ by sex (higher in male children on average) -- further validating the decision to standardize socioemotional functioning variables within each sex (performed in data wrangling above). 

## Check whether there are differences in diet covariates by adversity exposure
```{r}
#get relevant variables
diet_covs <- alpha_diversity_stress_usable %>% 
  dplyr::select(
    ctq_total_score_25it,
    STAI_prorated_state_pw26,
    Fibre.per.1000kcal,
    PUFA_.
  )

#correlations with adversity and diet covs
corPlot(diet_covs,
        numbers = TRUE,
        stars = TRUE,
        show.legend = TRUE,
        cex.axis = 0.8,
        cex = 0.5) 
#no associations between these diet covariates and adversity variables

#t-tests with categorical postnatal adversity variable
t.test(Fibre.per.1000kcal~postnatal_stress, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #n.s.
t.test(PUFA_.~postnatal_stress, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #n.s.
```

## Check that Sequencing Batch is not associated with other variables of interest
Test association between sequencing batch and adversity variables, and numbers in each batch
```{r eval=FALSE}
#t-test with sequencing batch and preconception, prenatal adversity
t.test(ctq_total_score_25it~Sequencing_batch, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(STAI_prorated_state_pw26~Sequencing_batch, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns

#chi-square test with sequencing batch and postnatal stress, adversity grouping variables
chisq.test(alpha_diversity_stress_usable$Sequencing_batch, alpha_diversity_stress_usable$postnatal_stress) #ns
chisq.test(alpha_diversity_stress_usable$Sequencing_batch, alpha_diversity_stress_usable$maltreated_mod_rev) #ns
chisq.test(alpha_diversity_stress_usable$Sequencing_batch, alpha_diversity_stress_usable$stai_above_clin_cutoff) #ns

#count the number of each group in the sequencing batches
alpha_diversity_stress_usable %>%
  filter(!is.na(postnatal_stress)) %>% 
  group_by(Sequencing_batch, postnatal_stress) %>%
  summarise(cnt = n()) %>%
  mutate(freq = round(cnt / sum(cnt), 3)) %>% 
  arrange(desc(freq)) #22% yes in batch1 (of non-nas), 25% in batch2

alpha_diversity_stress_usable %>%
  filter(!is.na(maltreated_mod_rev)) %>% 
  group_by(Sequencing_batch, maltreated_mod_rev) %>%
  summarise(cnt = n()) %>%
  mutate(freq = round(cnt / sum(cnt), 3)) %>% 
  arrange(desc(freq)) #42% yes in batch1 (of non-nas), 46% in batch2

alpha_diversity_stress_usable %>%
  filter(!is.na(stai_above_clin_cutoff)) %>% 
  group_by(Sequencing_batch, stai_above_clin_cutoff) %>%
  summarise(cnt = n()) %>%
  mutate(freq = round(cnt / sum(cnt), 3)) %>% 
  arrange(desc(freq)) #22% in batch 1 for each group
```
Conclusion: adversity grouping variable prevalences do not differ by sequencing batch 

# Initial Data Visualizations
## Barplots
```{r taxa barplots, warning=FALSE}
# get top 25 most abundant taxa
physeq_ttaxa <- get_top_taxa(physeq_obj = physeq_M24,
                             n = 25,
                             relative = TRUE,
                             discard_other = TRUE)

#group by postnatal adversity
physeq_ttaxa_leq <- merge_samples(physeq_ttaxa, "postnatal_stress")

#create 'genus species' variable
tmp_leq <- name_taxa(physeq_ttaxa_leq, species = TRUE)

#barplot based on postnatal adversity group (NAs are removed)
fantaxtic_bar(tmp_leq, color_by = "Species", label_by = "Species") +
  xlab("Postnatal Adversity") +
  ylab("Relative Abundance") +
  scale_fill_manual(values=as.vector(glasbey(32)))

#group by prenatal adversity group (above STAI-S cutoff for clinical levels of anxiety)
physeq_ttaxa_merged_stai <- merge_samples(physeq_ttaxa, "stai_state_above_clin_cutoff_pw26")
tmp_stai <- name_taxa(physeq_ttaxa_merged_stai, species = TRUE)
  
#barplot based on prenatal adversity (NAs are removed)
fantaxtic_bar(tmp_stai, color_by = "Species", label_by = "Species") +
  xlab("Prenatal Adversity") +
  ylab("Relative Abundance") +
  scale_fill_manual(values=as.vector(glasbey(32)))

#group by preconception adversity group ('moderate to severe' maltreatment in at least 1 category)
physeq_ttaxa_merged_ctq <- merge_samples(physeq_ttaxa, "maltreated_mod_rev")
tmp_ctq <- name_taxa(physeq_ttaxa_merged_ctq, species = TRUE)

#barplot based on preconception adversity
fantaxtic_bar(tmp_ctq, color_by = "Species", label_by = "Species") +
  xlab("Preconception Adversity") +
  ylab("Relative Abundance") +
  scale_fill_manual(values=as.vector(glasbey(32)))
```

## Distributions
```{r variable distributions, warning=FALSE}
# for LEQ, who was the respondent?
alpha_diversity_stress_usable %>% 
  group_by(any_bf_months) %>% 
  summarise(n())

#preconception adversity
alpha_diversity_stress_usable %>% 
  filter(timepoint==24) %>% 
  ggplot(aes(ctq_total_score_25it)) +
  geom_histogram(binwidth = 2) +
  xlab("Preconception Adversity Score")

#prenatal adversity
alpha_diversity_stress_usable %>% 
  filter(timepoint==24) %>% 
  ggplot(aes(STAI_prorated_state_pw26)) +
  geom_histogram(binwidth = 2) +
  geom_vline(aes(xintercept = 40, colour="cutoff")) +
  scale_color_manual(name = "Clinical Cutoff", values = c("cutoff" = "red"))

#postnatal adversity
alpha_diversity_stress_usable %>% 
  filter(timepoint==24 & !is.na(postnatal_stress)) %>% 
  ggplot(aes(postnatal_stress)) +
  geom_bar() +
  xlab("Postnatal Adversity")

#Alpha diversity -- shannon index
alpha_diversity_stress %>% 
  filter(timepoint==24) %>% 
  ggplot(aes(shannon_entropy)) +
  geom_histogram(binwidth = 0.25) +
  xlab("Shannon Index")

#alpha diveristy -- evenness
alpha_diversity_stress %>% 
  filter(timepoint==24) %>% 
  ggplot(aes(pielou_evenness)) +
  geom_histogram(binwidth = 0.05)

#alpha diversity -- faith pd
alpha_diversity_stress_usable%>% 
  filter(timepoint==24) %>% 
  ggplot(aes(faith_pd)) +
  geom_histogram(binwidth = 0.5) 

#alpha diversity -- observed features
alpha_diversity_stress_usable%>% 
  filter(timepoint==24) %>% 
  ggplot(aes(observed_features)) +
  geom_histogram(binwidth = 2)

#cbcl total problems age 2 years 
alpha_diversity_stress_usable %>%
  ggplot(aes(cbcltotprobtot_m24_std)) +
  geom_histogram(binwidth = 5) 

#cbcl sleep problems at age 4 years
alpha_diversity_stress_usable %>%
  ggplot(aes(cbclsleepprobtot_y4_std)) +
  geom_histogram(binwidth = 5) 
```

# Descriptive statistics
```{r eval=FALSE}
#Ns
#how many are of each ethnicity?
alpha_diversity_stress_usable %>% 
  dplyr::count(mother_ethnicity)
#each sex?
alpha_diversity_stress_usable %>% 
  dplyr::count(sex_binary) #0 = female, 1 = male
#each delivery mode?
alpha_diversity_stress_usable %>% 
  dplyr::count(deliv_mode_str)
#in each postnatal adveristy group?
alpha_diversity_stress_usable %>% 
  dplyr::count(postnatal_stress)
#in each preconception adversity group?
alpha_diversity_stress_usable %>% 
  dplyr::count(maltreated_mod_rev == 1)
#in each prenatal adveristy group?
alpha_diversity_stress_usable %>% 
  dplyr::count(STAI_prorated_state_pw26 > 40)
#each antibiotic use group? (yes/no)
alpha_diversity_stress_usable %>% 
  dplyr::count(antibio_M24 ==1)
#each probiotic use group? (yes/no)
alpha_diversity_stress_usable %>% 
  dplyr::count(probiotics_M24 ==1)
#each sequencing batch?
alpha_diversity_stress_usable %>% 
  dplyr::count(Sequencing_batch =="batch1")

#who is the respondent for LEQ?
alpha_diversity_stress_usable %>% 
  filter(!is.na(leq_b2_events_total_Y04)) %>%
  dplyr::count(person_filling_Y04)

#who is the respondent for CBCL year 2?
alpha_diversity_stress_usable %>% 
  filter(!is.na(cbcl_respondent_m24)) %>%
  dplyr::count(cbcl_respondent_m24)

#who is the respondent for CBCL year 4?
alpha_diversity_stress_usable %>% 
  filter(!is.na(cbcl_respondent_y4)) %>%
  dplyr::count(cbcl_respondent_y4)

#how many for each breastfeeding category? 
table(alpha_diversity_stress_usable$only_bf_months)
table(alpha_diversity_stress_usable$any_bf_months)

#mean, sd, range (continuous covariates, adversities, alpha diversity metrics)
favstats(~monthly_income_per_member, data=alpha_diversity_stress_usable)
favstats(~leq_b2_events_total_Y04, data=alpha_diversity_stress_usable)
favstats(~STAI_prorated_state_pw26, data=alpha_diversity_stress_usable)
favstats(~ctq_total_score_25it, data=alpha_diversity_stress_usable)
favstats(~GA, data=alpha_diversity_stress_usable)
favstats(~observed_features, data=alpha_diversity_stress_usable)
favstats(~faith_pd, data=alpha_diversity_stress_usable)
favstats(~shannon_entropy, data=alpha_diversity_stress_usable)
favstats(~pielou_evenness, data=alpha_diversity_stress_usable)

#mean, sd, range (cbcl variables)
favstats(~cbcldevprobtot_m24_std, data=alpha_diversity_stress_usable)
favstats(~cbclintprobtot_m24_std, data=alpha_diversity_stress_usable)
favstats(~cbclsleepprobtscore_m24, data=alpha_diversity_stress_usable)
favstats(~cbcltotprobtot_m24_std, data=alpha_diversity_stress_usable)
favstats(~cbcladhdprobtot_y4_std, data=alpha_diversity_stress_usable)
favstats(~cbclattprobtot_y4_std, data=alpha_diversity_stress_usable)
favstats(~cbcldevprobtot_y4_std, data=alpha_diversity_stress_usable)
favstats(~cbclextprobtot_y4_std, data=alpha_diversity_stress_usable)
favstats(~cbclintprobtot_y4_std, data=alpha_diversity_stress_usable)
favstats(~cbclsleepprobtot_y4_std, data=alpha_diversity_stress_usable)
favstats(~cbcltotprobtot_y4_std, data=alpha_diversity_stress_usable)

#how many complete on year 2 cbcl? (all participants that have 1 subscale, have all of them)
alpha_diversity_stress_usable %>% dplyr::count(!is.na(cbclintprobtot_m24_std)) 
alpha_diversity_stress_usable %>% dplyr::count(!is.na(cbclintprobtot_y4_std)) 

#mean, sd, range (age variables)
favstats(~c_age_years_stoolproxy_m24, data=alpha_diversity_stress_usable)
favstats(~age_y4_cbcl, data=alpha_diversity_stress_usable)
favstats(~age_m24_cbcl, data=alpha_diversity_stress_usable)

#how many complete on age variables?
alpha_diversity_stress_usable %>% dplyr::count(!is.na(age_m24_cbcl)) 
alpha_diversity_stress_usable %>% dplyr::count(!is.na(age_y4_cbcl)) 

#how many with complete data on all adversities?
alpha_diversity_stress_usable %>% dplyr::count(!is.na(postnatal_stress) & !is.na(ctq_total_score_25it) & !is.na(STAI_prorated_state_pw26)) 

#mean, sd, range (diet covariates)
favstats(~Fibre.per.1000kcal, data=alpha_diversity_stress_usable)
favstats(~PUFA_., data=alpha_diversity_stress_usable)
```

# Adversity Effects on the Gut Microbiome
## Alpha Diversity
### Covariate Selection: Bivariate Correlations with Continuous Covariates
```{r continuous covariate selection, warning=FALSE}
#create df with just continuous variables for correlation matrix
alpha_div_stress_cor <-
  alpha_diversity_stress_usable %>% 
  dplyr::select(
    shannon_entropy,
    faith_pd,
    pielou_evenness,
    observed_features,
    STAI_prorated_state_pw26,
    ctq_total_score_25it,
    monthly_income_per_member,
    GA,
    c_age_years_stoolproxy_m24, 
    `Protein_.energy`,
    `Fat_.energy`,
    `CHO_.energy`,
    Fibre.per.1000kcal,
    SatFat_.,
    MUFA_.,
    PUFA_.,
    DQI.score.adjusted
  )

#initialize file to save corplot
pdf(file = "../../figures/descriptives/Fig_S2.pdf")

#visualize correlations
corPlot(alpha_div_stress_cor,
        numbers=TRUE,
        main = "",
        stars=TRUE,
        labels = c("Shannon index", "Faith PD", "Pielou's Evennes", "Observed Features", "Prenatal Adversity", "Preconception Adversity", "Income", "Gestational Age", "Child Age at Stool Collection", "Protein", "Fat", "Carbs", "Fiber", "SatFat", "MUFA", "PUFA", "DQI"),
        show.legend=TRUE,
        diag=FALSE,
        upper=FALSE,
        scale = TRUE,
        xlas=2,
#        gr=gr,
        cex.axis = 0.6,
        cex = 0.4)

#save the plot
dev.off() 
 
#generate correlations and p-values for sig results
cor(alpha_div_stress_cor, use="pairwise.complete.obs")
cor.test(alpha_diversity_stress_usable$observed_features, alpha_diversity_stress_usable$monthly_income_per_member)
#fiber
cor.test(alpha_diversity_stress_usable$pielou_evenness, alpha_diversity_stress_usable$Fibre.per.1000kcal)
cor.test(alpha_diversity_stress_usable$shannon_entropy, alpha_diversity_stress_usable$Fibre.per.1000kcal)
#pufa
cor.test(alpha_diversity_stress_usable$pielou_evenness, alpha_diversity_stress_usable$PUFA_.)
cor.test(alpha_diversity_stress_usable$shannon_entropy, alpha_diversity_stress_usable$PUFA_.)
```
*Significant continuous covariates*: monthly income, Fiber, PUFA

### Covariate Selection: T-tests & ANOVAs with Categorical Covariates
Potential covariates: sex_binary, delivery mode, breastfeeding vars, ethnicity, antibiotics, probiotics, sequencing batch
```{r eval=FALSE}
#alpha diveristy by sex
t.test(shannon_entropy~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(faith_pd~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #females higher
t.test(pielou_evenness~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(observed_features~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #females higher

#differences in alpha div by delivery mode 
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  mutate(deliv_mode_str = as.factor(deliv_mode_str)) #convert categorical var to factor

t.test(shannon_entropy~deliv_mode_str, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(faith_pd~deliv_mode_str, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #vaginally born higher
t.test(pielou_evenness~deliv_mode_str, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns 
t.test(observed_features~deliv_mode_str, data = alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns


#difference in alpha div by duration of exclusive breastfeeding (one-way ANOVA)
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  mutate(only_bf_months = as.factor(only_bf_months)) #convert categorical var to factor

summary(aov(shannon_entropy~only_bf_months, data=alpha_diversity_stress_usable)) #ns
summary(aov(faith_pd~only_bf_months, data=alpha_diversity_stress_usable)) #ns
summary(aov(pielou_evenness~only_bf_months, data=alpha_diversity_stress_usable)) #ns
summary(aov(observed_features~only_bf_months, data=alpha_diversity_stress_usable)) #ns

#difference in alpha diversity by duration of any breastfeeding (one-way ANOVA)
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  mutate(any_bf_months = as.factor(any_bf_months)) #convert categorical var to factor

summary(aov(shannon_entropy~any_bf_months, data=alpha_diversity_stress_usable)) #ns
summary(aov(faith_pd~any_bf_months, data=alpha_diversity_stress_usable)) #ns
summary(aov(pielou_evenness~any_bf_months, data=alpha_diversity_stress_usable)) #ns
summary(aov(observed_features~any_bf_months, data=alpha_diversity_stress_usable)) #ns

#difference in alpha diversity by ethnicity (one-way ANOVA)
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  mutate(mother_ethnicity = as.factor(mother_ethnicity)) #convert categorical var to factor

summary(aov(shannon_entropy~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(faith_pd~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(pielou_evenness~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns 
summary(aov(observed_features~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns

#difference in alpha diversity by antibiotics
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  mutate(antibio_M24 = as.factor(antibio_M24))

t.test(shannon_entropy~antibio_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(faith_pd~antibio_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(pielou_evenness~antibio_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(observed_features~antibio_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns

#difference in alpha diversity by probiotics
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  mutate(probiotics_M24 = as.factor(probiotics_M24))

t.test(shannon_entropy~probiotics_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(faith_pd~probiotics_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(pielou_evenness~probiotics_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(observed_features~probiotics_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns

#difference in alpha diversity by sequencing batch
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  mutate(Sequencing_batch = as.factor(Sequencing_batch))

t.test(shannon_entropy~Sequencing_batch, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #sig (batch 1 higher)
t.test(faith_pd~Sequencing_batch, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(pielou_evenness~Sequencing_batch, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #sig (batch 1 higher)
t.test(observed_features~Sequencing_batch, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns

```
*Significant categorical covariates*: delivery mode, sex, sequencing batch

### Regression Prep: dummy code multicategorical variables
Non-binary categorical covariates: ethnicity (3 levels), duration of exclusive breastfeeding (4 levels)   
Dummy coding means each group is compared to the reference group
```{r results='hide'}
#dummy code ethnicity
alpha_diversity_stress_usable <- base::transform(alpha_diversity_stress_usable,
                       eth_f1 = (mother_ethnicity == "malay"),
                       eth_f2 = (mother_ethnicity == "indian"))
#factor 1 = Malay relative to Chinese
#factor 2 = Indian relative to Chinese 
#Chinese is reference group

#dummy code duration of exclusive breastfeeding
alpha_diversity_stress_usable <- base::transform(alpha_diversity_stress_usable,
                       bf_f1 = (only_bf_months == "1M_to_3M"),
                       bf_f2 = (only_bf_months == "3M_to_6M"),
                       bf_f3 = (only_bf_months == "6M_to_12M"))
#less than 1 month is reference group
#factor 1 = 1 to 3 months relative to 1 month
#factor 2 = 3 to 6 months relative to 1 month
#factor 3 = 6 to 12 months relative to 1 month

#dummy code cumulative adversity in comparison to no exposure group
alpha_diversity_stress_usable <- base::transform(alpha_diversity_stress_usable,
                       cumulative_0v1 = (cumulative_adv == "1"),
                       cumulative_0v2 = (cumulative_adv == "2"))
#0 (no adversity exposure) is the reference group
#factor 1 = no exposure relative to 1 timepoint
#factor 2 = no exposure relative to 2+ timepoints
```

### Regression Diagnostics
Note: We do not need to check normality of residuals. Since the number of observations per variable is > 10 in all models, violations of normality will not impact results [Schmidt & Finan, 2018](https://www.sciencedirect.com/science/article/pii/S0895435617304857)
```{r eval=FALSE}
#define base models -- no covariates
base_mod_allexp <- lm(shannon_entropy~ctq_total_score_25it +STAI_prorated_state_pw26 + postnatal_stress, data = alpha_diversity_stress_usable, na.action = na.exclude) #na.exclude allows for residuals to have NAs added so that they can be joined to df which includes more than complete cases
alpha_diversity_stress_usable$pred_base_mod_allexp <- predict(base_mod_allexp)
alpha_diversity_stress_usable$resid_base_mod_allexp <- residuals(base_mod_allexp)

base_mod_cumu <- lm(shannon_entropy~cumulative_0v1 + cumulative_0v2, data = alpha_diversity_stress_usable, na.action = na.exclude)
alpha_diversity_stress_usable$pred_base_mod_cumu <- predict(base_mod_cumu)
alpha_diversity_stress_usable$resid_base_mod_cumu <- residuals(base_mod_cumu)

#check homoskadisticity of residuals -- base models
gf_point(resid_base_mod_allexp~pred_base_mod_allexp, data = alpha_diversity_stress_usable) #heteroskadistic
gf_point(resid_base_mod_cumu~pred_base_mod_cumu, data = alpha_diversity_stress_usable) #pretty homoskadistic

#outliers
ols_plot_dfbetas(base_mod_allexp)
ols_plot_dfbetas(base_mod_cumu)

#define models with covariates (delivery mode, child sex, sequencing batch, income, fiber, PUFA)
cov1_mod_allexp <-lm(shannon_entropy~ctq_total_score_25it+STAI_prorated_state_pw26 + postnatal_stress+deliv_mode+sex_binary+monthly_income_per_member+Sequencing_batch+Fibre.per.1000kcal+PUFA_., data = alpha_diversity_stress_usable, na.action = na.exclude)
alpha_diversity_stress_usable$pred_cov1_mod_allexp <- predict(cov1_mod_allexp)
alpha_diversity_stress_usable$resid_cov1_mod_allexp <- residuals(cov1_mod_allexp)

cov1_mod_cumu <- lm(shannon_entropy~cumulative_0v1 + cumulative_0v2+deliv_mode+sex_binary+monthly_income_per_member+Sequencing_batch+Fibre.per.1000kcal+PUFA_., data = alpha_diversity_stress_usable, na.action = na.exclude)
alpha_diversity_stress_usable$pred_cov1_mod_cumu <- predict(cov1_mod_cumu)
alpha_diversity_stress_usable$resid_cov1_mod_cumu <- resid(cov1_mod_cumu)

#check homoskedasticity of residuals -- cov1 models
gf_point(resid_cov1_mod_allexp~pred_cov1_mod_allexp, data = alpha_diversity_stress_usable) #heteroskadistic
gf_point(pred_cov1_mod_cumu~pred_cov1_mod_cumu, data = alpha_diversity_stress_usable) #homoskadistic

#check linearity with respect to each predictor
gf_point(shannon_entropy~ctq_total_score_25it, data=alpha_diversity_stress_usable)
gf_point(shannon_entropy~STAI_prorated_state_pw26, data=alpha_diversity_stress_usable)
gf_violin(shannon_entropy~postnatal_stress, data=na.omit(alpha_diversity_stress_usable))
```
**Conclusions:** homoskadisticity of residuals is violated -- will perform regression with HC3 standard errors

### All Adversity Exposures
#### Shannon Index
```{r adversities shannon index, warning=FALSE}
#create complete cases dataset to use for both with and without covariates (to ensure same sample used in comparison)
alpha_div_all_cov_complete <- #create complete cases dataset for this analysis
  alpha_diversity_stress_usable %>% 
  filter(!is.na(ctq_total_score_25it) & !is.na(STAI_prorated_state_pw26) & !is.na(postnatal_stress) & !is.na(deliv_mode) & !is.na(sex_binary) &!is.na(monthly_income_per_member) &!is.na(Sequencing_batch) & !is.na(PUFA_.)) 

#define models 
cov1_mod_mult <- lm(shannon_entropy~ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress+sex_binary+deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_diversity_stress_usable)
base_mod_mult <- lm(shannon_entropy~ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress, data = alpha_div_all_cov_complete)

#run models (HC3 robust standard errors, showing standardized SE and coefficients, and saving to file)
knitr::knit_print(tab_model(cov1_mod_mult, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/shannon_covs.doc"))
#knitr::knit_print will print the results (which is automatically generated as a .html, and not printed to console)
knitr::knit_print(tab_model(base_mod_mult, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/shannon_ncovs.doc"))

#calculate effect sizes (eta squared/change in r squared) -- covariates
eta_shannon_covs <- with(alpha_div_all_cov_complete, spcor(cbind(ctq_total_score_25it, STAI_prorated_state_pw26, postnatal_stress,  sex_binary, deliv_mode, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., shannon_entropy))$estimate^2)
kable(eta_shannon_covs[,ncol(eta_shannon_covs)], col.names="R^2 of each variable on shannon index") %>% kable_styling() #output to console the column that has effect sizes we are interested in (shannon entropy as the outcome)

#calculate effect sizes (eta squared/change in r squared) -- no covariates
eta_shannon_ncovs <- with(alpha_div_all_cov_complete, spcor(cbind(ctq_total_score_25it, STAI_prorated_state_pw26, postnatal_stress, shannon_entropy))$estimate^2)
kable(eta_shannon_ncovs[,ncol(eta_shannon_ncovs)], col.names="R^2 of each variable on shannon index") %>% kable_styling() #kable functions give the output nicer formatting
```
#### Faith's PD
```{r, warning=FALSE}
#FAITH
#create complete cases dataset to use for both with and without covariates (to ensure same sample used in comparison)
alpha_div_faith_all_cov_complete <-
  alpha_diversity_stress_usable %>% 
  filter(!is.na(ctq_total_score_25it) & !is.na(STAI_prorated_state_pw26) & !is.na(postnatal_stress) & !is.na(deliv_mode) & !is.na(sex_binary) &!is.na(monthly_income_per_member)&!is.na(Sequencing_batch)& !is.na(PUFA_.))

#define models
cov1_mod_mult_faith <- lm(faith_pd~ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress+sex_binary+deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_diversity_stress_usable)
base_mod_mult_faith <- lm(faith_pd~ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress, data = alpha_div_faith_all_cov_complete)

#run models (HC3 robust standard errors, showing standardized SE and coefficients, and saving to file)
knitr::knit_print(tab_model(cov1_mod_mult_faith, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/faith_covs.doc"))
knitr::knit_print(tab_model(base_mod_mult_faith, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/faith_nocovs.doc"))

#find mean of faith pd by postnatal stress to see direction of effect
alpha_diversity_stress_usable %>% group_by(postnatal_stress) %>% summarise_at(vars(faith_pd), list(name = mean))

#calculate effect sizes (eta squared/change in r squared) -- covariates
eta_faith_covs <- with(alpha_div_faith_all_cov_complete, spcor(cbind(deliv_mode, sex_binary, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., ctq_total_score_25it, STAI_prorated_state_pw26, postnatal_stress, faith_pd))$estimate^2) 
kable(eta_faith_covs[,ncol(eta_faith_covs)], col.names="R^2 of each variable on Faith's PD") %>% kable_styling()

#calculate effect sizes (eta squared/change in r squared) -- no covariates
alpha_div_faith_all_base_complete <-
  alpha_diversity_stress_usable %>% 
  filter(!is.na(ctq_total_score_25it) & !is.na(STAI_prorated_state_pw26) & !is.na(postnatal_stress)) 

eta_faith_ncovs <- with(alpha_div_faith_all_cov_complete, spcor(cbind(ctq_total_score_25it, STAI_prorated_state_pw26, postnatal_stress, faith_pd))$estimate^2) 
kable(eta_faith_ncovs[,ncol(eta_faith_ncovs)], col.names="R^2 of each variable on Faith's PD") %>% kable_styling()
```
#### Observed Features
```{r, warning=FALSE}
#OBSERVED FEATURES
alpha_div_obs_all_cov_complete <-
  alpha_diversity_stress_usable %>% 
  filter(!is.na(ctq_total_score_25it) & !is.na(STAI_prorated_state_pw26) & !is.na(postnatal_stress) & !is.na(deliv_mode) & !is.na(sex_binary) &!is.na(monthly_income_per_member)&!is.na(Sequencing_batch)& !is.na(PUFA_.))

#define models
cov1_mod_mult_feat <- lm(observed_features~ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress+sex_binary+deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_diversity_stress_usable)
base_mod_mult_feat <- lm(observed_features~ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress, data = alpha_div_obs_all_cov_complete)

#run models (HC3 robust standard errors, showing standardized SE and coefficients, and saving to file)
knitr::knit_print(tab_model(cov1_mod_mult_feat, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/observed_covs.doc"))
knitr::knit_print(tab_model(base_mod_mult_feat, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/observed_nocovs.doc"))

#calculate effect sizes (eta squared/change in r squared) -- no covariates
eta_features_ncovs <- with(alpha_div_obs_all_cov_complete, spcor(cbind(ctq_total_score_25it, STAI_prorated_state_pw26, postnatal_stress, observed_features))$estimate^2)
kable(eta_features_ncovs[,ncol(eta_features_ncovs)], col.names="R^2 of each variable on Observed Features") %>% kable_styling()

#calculate effect sizes (eta squared/change in r squared) -- covariates
eta_features_covs <- with(alpha_div_obs_all_cov_complete, spcor(cbind(deliv_mode, sex_binary, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., ctq_total_score_25it, STAI_prorated_state_pw26, postnatal_stress, observed_features))$estimate^2) 
kable(eta_features_covs[,ncol(eta_features_covs)], col.names="R^2 of each variable on Observed Features") %>% kable_styling()
```
#### Pielou Evenness
```{r, warning=FALSE}
#EVENNESS
#define models
cov1_mod_mult_even <- lm(pielou_evenness~ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress+sex_binary+deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_diversity_stress_usable)
 base_mod_mult_even <- lm(pielou_evenness~ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress, data = alpha_div_obs_all_cov_complete)

#run models (HC3 robust standard errors, showing standardized SE and coefficients, and saving to file)
knitr::knit_print(tab_model(cov1_mod_mult_even, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/evenness_covs.doc"))
knitr::knit_print(tab_model(base_mod_mult_even, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/evenness_nocovs.doc"))

#calculate effect sizes (eta squared/change in r squared) --  no covariates
eta_even_ncovs <- with(alpha_div_obs_all_cov_complete, spcor(cbind(ctq_total_score_25it, STAI_prorated_state_pw26, postnatal_stress, pielou_evenness))$estimate^2) 
kable(eta_even_ncovs[,ncol(eta_even_ncovs)], col.names="R^2 of each variable on Pielou Evenness") %>% kable_styling()

#calculate effect sizes (eta squared/change in r squared) -- covariates
eta_even_covs <- with(alpha_div_obs_all_cov_complete, spcor(cbind(deliv_mode, sex_binary, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., ctq_total_score_25it, STAI_prorated_state_pw26, postnatal_stress, pielou_evenness))$estimate^2) 
kable(eta_even_covs[,ncol(eta_even_covs)], col.names="R^2 of each variable on Pielou Evenness") %>% kable_styling()
```
### Serial mediation to quantify all direct and indirect effects of adversities
Reviewer asked to quantify the relative effects of each timepoint of adversity

#### Prepare data for use in Mplus
Need to specify serial mediation models in Mplus due to the inclusion of a binary mediator
```{r}
#filter for only complete cases on all variables used in this set of analyses 
alpha_div_stress_shannon_serial <- alpha_diversity_stress_usable %>% 
  filter(!is.na(ctq_total_score_25it) & !is.na(STAI_prorated_state_pw26) & !is.na(postnatal_stress) & !is.na(monthly_income_per_member) & !is.na(deliv_mode)& !is.na(Fibre.per.1000kcal) & !is.na(PUFA_.)) 
#rename variables to be short and output .dat file for mplus
serial_mplus <- alpha_div_stress_shannon_serial %>% 
  dplyr::select(
    '#SampleID',
    precon = ctq_total_score_25it,
    prenat = STAI_prorated_state_pw26,
    postnatal_stress,
    sex_binary,
    deliv_mode,
    income = monthly_income_per_member,
    Sequencing_batch,
    fiber = Fibre.per.1000kcal,
    pufa = PUFA_.,
    shannon = shannon_entropy,
    faith = faith_pd,
    feat = observed_features,
    even = pielou_evenness
  ) %>% 
  mutate(
    id = seq(1, nrow(alpha_div_stress_shannon_serial), by=1), #create a numbered list for id variable
    postnat = as.numeric(as.character(postnatal_stress)),
    sex = as.numeric(as.character(sex_binary)),
    deliv = as.numeric(as.character(deliv_mode)),
    seq = ifelse(Sequencing_batch=="batch1", 1, 0)
  ) %>% 
  dplyr::select(
    -'#SampleID',
    -postnatal_stress,
    -sex_binary,
    -deliv_mode,
    -Sequencing_batch
  )

write.dat(serial_mplus, "../../data/analysis_datasets/fran/serial_mediation_alpha_div/mplus_serial.dat/gusto_mplus_revctq/mplus_serial_v2.dat")
```

#### Serial Mediation Models in Mplus
See gusto_serial_mediation Mplus script files (".inp") for serial mediation syntax in Mplus, and gusto_serial_mediation_totaleffects Mplus script files for specifications of total effects of each adversity exposure in Mplus. Since with binary mediators the total effect may not be equal to the sum of the direct and indirect effects, we calculated total effects as simply the effect of each adversity without the others in the model. 

Mplus version 8.3 was used for all analyses. 

#### Calculate R^2 for total effects models
Total effects models were calculated in Mplus to be consistent with modeling for the serial mediation models, but change in R^2 for predictor of interest was calculated in R. 
```{r}
#shannon, preconception total effects model
total_shannon_precon = lm(shannon_entropy ~ ctq_total_score_25it+sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch + Fibre.per.1000kcal + PUFA_., data=alpha_div_all_cov_complete)

eta_shannon_precon <- with(alpha_div_all_cov_complete, spcor(cbind(ctq_total_score_25it, sex_binary, deliv_mode, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., shannon_entropy))$estimate^2)
kable(eta_shannon_precon[,ncol(eta_shannon_precon)], col.names="R^2 of each variable on Shannon index") %>% kable_styling()

#shannon, prenatal total effects model
total_shannon_prenat = lm(shannon_entropy ~ STAI_prorated_state_pw26+sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch + Fibre.per.1000kcal + PUFA_., data=alpha_div_all_cov_complete)

eta_shannon_prenat <- with(alpha_div_all_cov_complete, spcor(cbind(STAI_prorated_state_pw26, sex_binary, deliv_mode, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., shannon_entropy))$estimate^2)
kable(eta_shannon_prenat[,ncol(eta_shannon_prenat)], col.names="R^2 of each variable on Shannon index") %>% kable_styling()

#faith, preconception total effects model
total_faith_precon = lm(faith_pd ~ ctq_total_score_25it+sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch + Fibre.per.1000kcal + PUFA_., data=alpha_div_all_cov_complete)

eta_faith_precon <- with(alpha_div_all_cov_complete, spcor(cbind(ctq_total_score_25it, sex_binary, deliv_mode, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., faith_pd))$estimate^2)
kable(eta_faith_precon[,ncol(eta_faith_precon)], col.names="R^2 of each variable on Faith's PD") %>% kable_styling()

#faith, prenatal total effects model
total_faith_prenat = lm(faith_pd ~ STAI_prorated_state_pw26+sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch + Fibre.per.1000kcal + PUFA_., data=alpha_div_all_cov_complete)

eta_faith_prenat <- with(alpha_div_all_cov_complete, spcor(cbind(STAI_prorated_state_pw26, sex_binary, deliv_mode, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., faith_pd))$estimate^2)
kable(eta_faith_prenat[,ncol(eta_faith_prenat)], col.names="R^2 of each variable on Faith's PD") %>% kable_styling()

#observed features, preconception total effects model
total_feat_precon = lm(observed_features ~ ctq_total_score_25it+sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch + Fibre.per.1000kcal + PUFA_., data=alpha_div_all_cov_complete)

eta_feat_precon <- with(alpha_div_all_cov_complete, spcor(cbind(ctq_total_score_25it, sex_binary, deliv_mode, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., observed_features))$estimate^2)
kable(eta_feat_precon[,ncol(eta_feat_precon)], col.names="R^2 of each variable on Obs. Features") %>% kable_styling()

#observed features, prenatal total effects model
total_feat_prenat = lm(observed_features ~ STAI_prorated_state_pw26+sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch + Fibre.per.1000kcal + PUFA_., data=alpha_div_all_cov_complete)

eta_feat_prenat <- with(alpha_div_all_cov_complete, spcor(cbind(STAI_prorated_state_pw26, sex_binary, deliv_mode, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., observed_features))$estimate^2)
kable(eta_feat_prenat[,ncol(eta_feat_prenat)], col.names="R^2 of each variable on Obs. Features") %>% kable_styling()

#evenness, preconception total effects model
total_even_precon = lm(pielou_evenness ~ ctq_total_score_25it+sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch + Fibre.per.1000kcal + PUFA_., data=alpha_div_all_cov_complete)

eta_even_precon <- with(alpha_div_all_cov_complete, spcor(cbind(ctq_total_score_25it, sex_binary, deliv_mode, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., pielou_evenness))$estimate^2)
kable(eta_even_precon[,ncol(eta_even_precon)], col.names="R^2 of each variable on Pielou evenness") %>% kable_styling()

#evenness, prenatal total effects model
total_even_prenatal = lm(pielou_evenness ~ STAI_prorated_state_pw26+sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch + Fibre.per.1000kcal + PUFA_., data=alpha_div_all_cov_complete)

#calculate change in R^2 for prenatal adversity
eta_even_prenat <- with(alpha_div_all_cov_complete, spcor(cbind(STAI_prorated_state_pw26, sex_binary, deliv_mode, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_., pielou_evenness))$estimate^2)
kable(eta_even_prenat[,ncol(eta_even_prenat)], col.names="R^2 of each variable on Pielou evenness") %>% kable_styling()
```

### Cumulative Adversity 
#### Shannon Index
```{r, warning=FALSE}
#SHANNON INDEX
y <- favstats(shannon_entropy~cumulative_adv, data=alpha_diversity_stress_usable) #get group sizes

#first, create dataset of complete cases 
alpha_div_cumulative_complete <-
  alpha_diversity_stress_usable %>% 
  filter(!is.na(cumulative_adv) & !is.na(deliv_mode) & !is.na(sex_binary) &!is.na(monthly_income_per_member) & !is.na(Sequencing_batch) & !is.na(PUFA_.)) 

#define models
base_mod_cumulative0v2_sum <- lm(shannon_entropy~cumulative_0v1 + cumulative_0v2, data = alpha_div_cumulative_complete)
cov1_mod_cumulative0v2_sum <- lm(shannon_entropy~cumulative_0v1 + cumulative_0v2+deliv_mode+sex_binary+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_diversity_stress_usable)

#run models (HC3 robust standard errors, showing standardized SE and coefficients, and saving to file)
knitr::knit_print(tab_model(cov1_mod_cumulative0v2_sum, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/shannon_cumulative_covs.doc"))
knitr::knit_print(tab_model(base_mod_cumulative0v2_sum, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/shannon_cumulative_nocovs.doc"))

#calculate effect sizes (eta squared/change in r squared) -- covariates
eta_cumul_shannon_covs <- with(alpha_div_cumulative_complete, spcor(cbind(cumulative_0v1, cumulative_0v2, deliv_mode, sex_binary, monthly_income_per_member, Fibre.per.1000kcal, PUFA_., Sequencing_batch, shannon_entropy))$estimate^2) 
kable(eta_cumul_shannon_covs[,ncol(eta_cumul_shannon_covs)], col.names="R^2 of each variable on Shannon Index") %>% kable_styling()

#calculate effect sizes (eta squared/ change in r squared) -- no covariates
eta_cumul_shannon_ncovs <- with(alpha_div_cumulative_complete, spcor(cbind(cumulative_0v1, cumulative_0v2, shannon_entropy))$estimate^2)
kable(eta_cumul_shannon_ncovs[,ncol(eta_cumul_shannon_ncovs)], col.names="R^2 of each variable on Shannon Index") %>% kable_styling()
```
#### Faith's PD
```{r, warning=FALSE}
#FAITH PD
#define models
base_mod_cumulative0v2_faith <- lm(faith_pd~cumulative_0v1 + cumulative_0v2, data = alpha_div_cumulative_complete)
cov1_mod_cumulative0v2_faith <- lm(faith_pd~cumulative_0v1 + cumulative_0v2+sex_binary + deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_diversity_stress_usable)

#run models (HC3 robust standard errors, showing standardized SE and coefficients, and saving to file)
knitr::knit_print(tab_model(cov1_mod_cumulative0v2_faith, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/faith_cumulative_covs.doc"))
knitr::knit_print(tab_model(base_mod_cumulative0v2_faith, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/faith_cumulative_nocovs.doc"))

#calculate effect sizes (eta squared/change in r squared) -- covariates
eta_cumul_faith_covs <- with(alpha_div_cumulative_complete, spcor(cbind(cumulative_0v1, cumulative_0v2, deliv_mode, sex_binary, monthly_income_per_member, Sequencing_batch,Fibre.per.1000kcal, PUFA_., faith_pd))$estimate^2) 
kable(eta_cumul_faith_covs[,ncol(eta_cumul_faith_covs)], col.names="R^2 of each variable on Faith's PD") %>% kable_styling()

#calculate effect sizes (eta squared/change in r squared) -- no covariates
eta_cumul_faith_ncovs <- with(alpha_div_cumulative_complete, spcor(cbind(cumulative_0v1, cumulative_0v2, faith_pd))$estimate^2)
kable(eta_cumul_faith_ncovs[,ncol(eta_cumul_faith_ncovs)], col.names="R^2 of each variable on Faith's PD") %>% kable_styling()
```
#### Observed Features
```{r, warning=FALSE}
#OBSERVED FEATURES
#define models
base_mod_cumulative_feat <- lm(observed_features~cumulative_0v1 + cumulative_0v2, data = alpha_div_cumulative_complete)
cov1_mod_cumulative_feat <- lm(observed_features~cumulative_0v1 + cumulative_0v2+sex_binary + deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_diversity_stress_usable)

#run models (HC3 robust standard errors, showing standardized SE and coefficients, and saving to file)
knitr::knit_print(tab_model(cov1_mod_cumulative_feat, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/observed_cumulative_covs.doc"))
knitr::knit_print(tab_model(base_mod_cumulative_feat, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/observed_cumulative_nocovs.doc"))

#calculate effect sizes (eta squared/change in r squared) -- covariates
eta_cumul_feat_covs <- with(alpha_div_cumulative_complete, spcor(cbind(cumulative_0v1, cumulative_0v2, deliv_mode, sex_binary, monthly_income_per_member, Sequencing_batch,Fibre.per.1000kcal, PUFA_., observed_features))$estimate^2) 
 kable(eta_cumul_feat_covs[,ncol(eta_cumul_feat_covs)], col.names="R^2 of each variable on Observed Features") %>% kable_styling()

#calculate effect sizes (eta squared/change in r squared) -- no covariates
eta_cumul_feat_ncovs <- with(alpha_div_cumulative_complete, spcor(cbind(cumulative_0v1, cumulative_0v2, observed_features))$estimate^2)
kable(eta_cumul_feat_ncovs[,ncol(eta_cumul_feat_ncovs)], col.names="R^2 of each variable on Observed Features") %>% kable_styling()
```
#### Pielou Evenness
```{r, warning=FALSE}
#PIELOU EVENNESS
#define models
base_mod_cumulative0v2_even <- lm(pielou_evenness~cumulative_0v1 + cumulative_0v2, data = alpha_div_cumulative_complete)
cov1_mod_cumulative0v2_even <- lm(pielou_evenness~cumulative_0v1 + cumulative_0v2+sex_binary + deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_diversity_stress_usable)

#run models (HC3 robust standard errors, showing standardized SE and coefficients, and saving to file)
knitr::knit_print(tab_model(base_mod_cumulative0v2_even, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/even_cumulative_nocovs.doc"))
knitr::knit_print(tab_model(cov1_mod_cumulative0v2_even, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file="../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/even_cumulative_covs.doc"))

#calculate effect sizes (eta squared/change in r squared) -- covariates
eta_cumul_even_covs <- with(alpha_div_cumulative_complete, spcor(cbind(cumulative_0v1, cumulative_0v2, deliv_mode, sex_binary, monthly_income_per_member, Sequencing_batch, Fibre.per.1000kcal, PUFA_.,pielou_evenness))$estimate^2) 
kable(eta_cumul_even_covs[,ncol(eta_cumul_even_covs)], col.names="R^2 of each variable on Pielou Evenness") %>% kable_styling()

#calculate effect sizes (eta squared/change in r squared) -- no covariates
eta_cumul_even_ncovs <- with(alpha_div_cumulative_complete, spcor(cbind(cumulative_0v1, cumulative_0v2, pielou_evenness))$estimate^2)
kable(eta_cumul_even_ncovs[,ncol(eta_cumul_even_ncovs)], col.names="R^2 of each variable on Pielou Evenness") %>% kable_styling()
```

## Alpha Diversity Visualizations
### Scatterplots & Boxplots: Alpha Diversity by Adversity
```{r ELA alpha div graphs, warning=FALSE}
#postnatal adversity
d_postnatal <- alpha_diversity_stress_usable %>% dplyr::select(subjid, postnatal_stress, shannon_entropy, faith_pd, pielou_evenness, observed_features) #grab only needed vars
# Than transform the data to long format as stated already in the comments using reshape function melt
d_postnatal <- d_postnatal %>% 
  mutate(
    postnatal_stress_s = case_when(
      postnatal_stress==1 ~ "yes",
      postnatal_stress==0 ~ "no"
    ) 
  ) %>% 
  dplyr::select(
      -postnatal_stress
    )

d_postnatal_long <- melt(d_postnatal)
d_postnatal_long_complete <- d_postnatal_long %>% filter(across(.cols=everything(), ~!is.na(.x))) #create complete dataset

#Boxplots
ggplot(d_postnatal_long_complete, aes(y = value, x = postnatal_stress_s, colour = postnatal_stress_s)) + 
  geom_jitter() +
  geom_boxplot(alpha=0.3, outlier.shape=NA) +
  xlab("Postnatal Stress Exposure") + 
  ylab("Value") +
  facet_wrap( ~ variable, scales = "free", labeller= as_labeller(c('shannon_entropy' = "Shannon's Index", 'faith_pd' = "Faith's PD", 'pielou_evenness' = "Pielou Evenness", 'observed_features' = "Observed Features"))) + 
  theme_bw() +
  stat_summary(fun=mean, colour='royalblue3', geom='point', shape='-', size=10) +
  scale_color_discrete("Postnatal Stress Exposure", labels = c("no (N = 233)", "yes (N= 76)")) 

#Prenatal Adversity
grid.arrange(ggplot(alpha_diversity_stress_usable, aes(STAI_prorated_state_pw26, shannon_entropy))+geom_point(size=1)+geom_smooth(method="lm") + labs(x="Prenatal Adversity Score", y= "Shannon Index"),
             ggplot(alpha_diversity_stress_usable, aes(STAI_prorated_state_pw26, faith_pd))+geom_point(size=1)+geom_smooth(method="lm")+ labs(x="Prenatal Adversity Score", y="Faith's PD"),
             ggplot(alpha_diversity_stress_usable, aes(STAI_prorated_state_pw26, pielou_evenness))+geom_point(size=1)+geom_smooth(method="lm")+ labs(x="Prenatal Adversity Score", y="Pielou Evenness"),
             ggplot(alpha_diversity_stress_usable, aes(STAI_prorated_state_pw26, observed_features))+geom_point(size=1)+geom_smooth(method="lm")+ labs(x="Prenatal Adversity Score", y="Observed Features"))

#Preconception adversity
grid.arrange(ggplot(alpha_diversity_stress_usable, aes(ctq_total_score_25it, shannon_entropy))+geom_point(size=1)+geom_smooth(method="lm") + labs(x="Preconception Adversity Score", y= "Shannon Index"),
             ggplot(alpha_diversity_stress_usable, aes(ctq_total_score_25it, faith_pd))+geom_point(size=1)+geom_smooth(method="lm")+ labs(x="Prenatal Adversity Score", y="Faith's PD"),
             ggplot(alpha_diversity_stress_usable, aes(ctq_total_score_25it, pielou_evenness))+geom_point(size=1)+geom_smooth(method="lm")+ labs(x="Preconception Adversity Score", y="Pielou Evenness"),
             ggplot(alpha_diversity_stress_usable, aes(ctq_total_score_25it, observed_features))+geom_point(size=1)+geom_smooth(method="lm")+ labs(x="Preconception Adversity Score", y="Observed Features"))

#cumulative adversity
d_cumulative <- alpha_diversity_stress_usable %>% dplyr::select(subjid, cumulative_adv, shannon_entropy, faith_pd, pielou_evenness, observed_features) 
d_cumulative_long <- melt(d_cumulative)
d_cumulative_long_complete <- d_cumulative_long %>% filter(across(.cols=everything(), ~!is.na(.x))) #create complete dataset

#Boxplots
ggplot(d_cumulative_long_complete, aes(y = value, x = cumulative_adv, colour = cumulative_adv)) + 
  geom_jitter() +
  geom_boxplot(alpha=0.3, outlier.shape=NA) +
  xlab("Cumulative Adversity Exposure") + 
  ylab("Value") +
  facet_wrap( ~ variable, scales = "free", labeller= as_labeller(c('shannon_entropy' = "Shannon Index", 'faith_pd' = "Faith's PD", 'pielou_evenness' = "Pielou Evenness", 'observed_features' = "Observed Features"))) + 
  theme_bw() +
  stat_summary(fun=mean, colour='royalblue3', geom='point', shape='-', size=10) +
  scale_color_discrete("Cumulative Adversity Exposure", labels = c("0 timepoints", "1 timepoint", "2+ timepoints")) 
```
### Figure 1:  Faith’s PD is Lower Among Children with Higher Postnatal Adversity, Evenness is Higher Among Children with Higher Prenatal Adversity 
```{r faith by postnatal, warning=FALSE}

#alpha_diversity_stress_usable <- read_csv("../../data/analysis_datasets/fran/alpha_div_stress_usable_ctqrev.csv")

#create group means dataset for a quick look at magnitude of effect
faith_means <- alpha_diversity_stress_usable %>% 
  filter(!is.na(postnatal_stress)) %>% 
        group_by(postnatal_stress) %>% 
        summarise(faith_pd = mean(faith_pd))

#add postnatal stress string variable to make plotting easier
alpha_diversity_stress_usable <-
  alpha_diversity_stress_usable %>% 
  mutate(
    postnatal_stress_str = case_when(
      postnatal_stress == 0 ~ "no",
      postnatal_stress == 1 ~ "yes"
    )
  )

#create residual for faith pd after removing variance associated with covariates
faith_resid <- lm(faith_pd ~Fibre.per.1000kcal +sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch+ctq_total_score_25it+STAI_prorated_state_pw26, data = alpha_diversity_stress_usable)

alpha_diversity_stress_usable_faith <- 
  alpha_diversity_stress_usable %>% 
  filter(!is.na(sex_binary) & !is.na(deliv_mode) &!is.na(monthly_income_per_member)&!is.na(Sequencing_batch) & !is.na(ctq_total_score_25it) & !is.na(STAI_prorated_state_pw26) & !is.na(Fibre.per.1000kcal)) %>% 
  mutate(
    faith_resid = resid(faith_resid)
  ) 

#determine Ns for the graph labels
alpha_diversity_stress_usable_faith %>% dplyr::count(postnatal_stress==1) #Yes: 104, No: 45

#boxplot with faith by postnatal adversity (partial correlations)
faith <- ggplot(alpha_diversity_stress_usable_faith %>% filter(!is.na(postnatal_stress_str)), aes(y = faith_resid, x = postnatal_stress_str, color = postnatal_stress_str)) +
geom_point(position=position_jitter(w=0.3, h=0)) +
geom_boxplot(alpha=0.3, outlier.shape=NA) +
xlab("Postnatal Adversity Exposure") + 
  scale_x_discrete("Postnatal Adversity Exposure", labels = c("no" = "No (N=104)", "yes" = "Yes (N=45)")
  ) +
ylab("Faith's Phylogenetic Diversity (residuals)") +
stat_summary(fun=mean, colour='#E7298A', geom='point', shape='-', size=20) + #the color is the hexadecimal for the 3rd color on dark2 palette
  scale_fill_brewer(palette = "Dark2") + #specify colorblind friendly color palette for fill
  scale_color_brewer(palette = "Dark2") + #specify colorblind friendly color palette for color
theme_minimal() +
theme(legend.position="none", #hide legend
  panel.grid.major.x = element_blank(), #hide grid lines
  panel.grid.minor.x = element_blank(),
  panel.grid.major.y = element_blank(),
  panel.grid.minor.y = element_blank(),
) +
  ggtitle('B')

#visualize colors from dark2 palette
display.brewer.pal(n = 8, name = 'Dark2')
brewer.pal(n = 8, name = "Dark2") #get hexadecimal names for each color

#create residual for pielou evenness after removing variance associated with covariates
even_resid <- lm(pielou_evenness ~Fibre.per.1000kcal +sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch+ctq_total_score_25it+postnatal_stress, data = alpha_diversity_stress_usable)

alpha_diversity_stress_usable_even <- 
  alpha_diversity_stress_usable %>% 
  filter(!is.na(sex_binary) & !is.na(deliv_mode) &!is.na(monthly_income_per_member)&!is.na(Sequencing_batch) & !is.na(ctq_total_score_25it) & !is.na(postnatal_stress) & !is.na(Fibre.per.1000kcal)) %>% 
  mutate(
    even_resid = resid(even_resid)
  ) 

#create prenatal adversity--eveness scatterplot
even <- ggplot(alpha_diversity_stress_usable_even %>% filter(!is.na(STAI_prorated_state_pw26)), aes(y = even_resid, x = STAI_prorated_state_pw26)) +
  geom_point(alpha=0.7, color='#7570B3') + #third color from dark2 colorbrewer palette
  geom_smooth(method="lm", color='#E7298A') + #fourth color from dark2 colorbrewer palette
  xlab("Prenatal Adversity Exposure") + 
ylab("Pielou Evenness (residuals)") +
  ylim(-0.2, 0.2) + 
theme_minimal() +
theme(legend.position="none", #hide legend
  panel.grid.major.x = element_blank(), #hide grid lines
  panel.grid.minor.x = element_blank(),
  panel.grid.major.y = element_blank(),
  panel.grid.minor.y = element_blank(),
) +
  ggtitle('A')

#visualize the two plots side by side
even + faith

#save graph with Faith PD residuals for inclusion in manuscript
ggsave("../../manuscripts/fran_adversity_microbiome/figures/figure1.svg", height=4, width=6, dpi=1000)

```

## Beta Diversity
Qiime forum with explanation of PERMANOVA and permdisp tests:    https://forum.qiime2.org/t/understanding-beta-group-significance-permanova-results/12648/4   

* Permanova asks if there is a difference in either within or between group distance for any of the groups. 
* Permdisp tests the hypothesis that there is a significant difference in within group variance.
* So, if you see a large signal in permanova and a small one in permdisp, that suggests that the difference is driven by differences between communities compared to differences within one of the communities.  
  
Info about effect size in adonis (R^2): http://qiime.org/tutorials/category_comparison.html   

### Covariate Selection
Potential covariates: sex_binary, delivery mode, breastfeeding vars, ethnicity, antibiotics, probiotics, sequencing batch

#### Child Sex
```{r, warning=FALSE}
physeq_M24_sex <- subset_samples(physeq_M24, sex_binary != "NA") #create a dataset that is complete on sex; necessary for distance matrix calculations to work

#get distance matrices (weighted unifrac, unweighted unifrac, bray-curtis, jaccard)
set.seed(main.seed) #need to run this right before all Unifrac distance matrices calculations 
M24_wunifrac_sex <- phyloseq::distance(physeq_M24_sex, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_sex <- phyloseq::distance(physeq_M24_sex, method = "unifrac", type = "samples")
M24_jaccard_sex <- phyloseq::distance(physeq_M24_sex, method = "jaccard", type = "samples")
M24_bc_sex <- phyloseq::distance(physeq_M24_sex, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_sex <- data.frame(sample_data(physeq_M24_sex)) #get data frame of metadata

#perform PERMANOVAs with each distance matrix
set.seed(main.seed) #need to run this before Unifrac distance matrices calculations 
test <- adonis(M24_wunifrac_sex~sex_binary, data=metadata_physeq_M24_sex)#ns
set.seed(main.seed)
adonis(M24_unifrac_sex~sex_binary, data=metadata_physeq_M24_sex) #ns
adonis(M24_jaccard_sex~sex_binary, data=metadata_physeq_M24_sex) #ns
adonis(M24_bc_sex~sex_binary, data=metadata_physeq_M24_sex) #ns
```
#### Antibiotic Use
```{r, warning=FALSE}
#ANTIBIOTICS
physeq_M24_antibio <- subset_samples(physeq_M24, antibio_M24 != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_antibio <- phyloseq::distance(physeq_M24_antibio, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_antibio <- phyloseq::distance(physeq_M24_antibio, method = "unifrac", type = "samples")
M24_jaccard_antibio <- phyloseq::distance(physeq_M24_antibio, method = "jaccard", type = "samples")
M24_bc_antibio <- phyloseq::distance(physeq_M24_antibio, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_antibio <- data.frame(sample_data(physeq_M24_antibio))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_antibio~antibio_M24, data=metadata_physeq_M24_antibio) #sig
adonis(M24_unifrac_antibio~antibio_M24, data=metadata_physeq_M24_antibio) #ns
adonis(M24_jaccard_antibio~antibio_M24, data=metadata_physeq_M24_antibio) #ns
adonis(M24_bc_antibio~antibio_M24, data=metadata_physeq_M24_antibio) #ns
```
#### Probiotic Use
```{r, warning=FALSE}
#PROBIOTICS
physeq_M24_probio <- subset_samples(physeq_M24, probiotics_M24 != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_probio <- phyloseq::distance(physeq_M24_probio, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_probio <- phyloseq::distance(physeq_M24_probio, method = "unifrac", type = "samples")
M24_jaccard_probio <- phyloseq::distance(physeq_M24_probio, method = "jaccard", type = "samples")
M24_bc_probio <- phyloseq::distance(physeq_M24_probio, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_probio <- data.frame(sample_data(physeq_M24_probio))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_probio~probiotics_M24, data=metadata_physeq_M24_probio) #ns
adonis(M24_unifrac_probio~probiotics_M24, data=metadata_physeq_M24_probio) #ns
adonis(M24_jaccard_probio~probiotics_M24, data=metadata_physeq_M24_probio) #ns
adonis(M24_bc_probio~probiotics_M24, data=metadata_physeq_M24_probio) #ns
```
#### Delivery Mode
```{r, warning=FALSE}
#DELIVERY MODE 
physeq_M24_dm <- subset_samples(physeq_M24, deliv_mode != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_dm <- phyloseq::distance(physeq_M24_dm, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_dm <- phyloseq::distance(physeq_M24_dm, method = "unifrac", type = "samples")
M24_jaccard_dm <- phyloseq::distance(physeq_M24_dm, method = "jaccard", type = "samples")
M24_bc_dm <- phyloseq::distance(physeq_M24_dm, method = "bray", type = "samples")
#get dataframe of metadata
metadata_physeq_M24_dm <- data.frame(sample_data(physeq_M24_dm))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_dm~deliv_mode, data=metadata_physeq_M24_dm)#ns
set.seed(main.seed)
adonis(M24_unifrac_dm~deliv_mode, data=metadata_physeq_M24_dm) #significant
adonis(M24_jaccard_dm~deliv_mode, data=metadata_physeq_M24_dm) #ns
adonis(M24_bc_dm~deliv_mode, data=metadata_physeq_M24_dm) #ns
```
#### Child Ethnicity
```{r, warning=FALSE}
#ETHNICITY
physeq_M24_eth <- subset_samples(physeq_M24, mother_ethnicity != "NA")

#calculate distance matrices
set.seed(main.seed)  
M24_wunifrac_eth <- phyloseq::distance(physeq_M24_eth, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_eth <- phyloseq::distance(physeq_M24_eth, method = "unifrac", type = "samples")
M24_jaccard_eth <- phyloseq::distance(physeq_M24_eth, method = "jaccard", type = "samples")
M24_bc_eth <- phyloseq::distance(physeq_M24_eth, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_eth <- data.frame(sample_data(physeq_M24_eth))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_eth~mother_ethnicity, data=metadata_physeq_M24_eth) #ns
adonis(M24_unifrac_eth~mother_ethnicity, data=metadata_physeq_M24_eth) #significant
adonis(M24_jaccard_eth~mother_ethnicity, data=metadata_physeq_M24_eth) #ns
adonis(M24_bc_eth~mother_ethnicity, data=metadata_physeq_M24_eth) #significant
```
#### Sequencing Batch
```{r, warning=FALSE}
#SEQUENCING BATCH
physeq_M24_seqba <- subset_samples(physeq_M24, Sequencing_batch != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_seqba <- phyloseq::distance(physeq_M24_seqba, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_seqba <- phyloseq::distance(physeq_M24_seqba, method = "unifrac", type = "samples")
M24_jaccard_seqba <- phyloseq::distance(physeq_M24_seqba, method = "jaccard", type = "samples")
M24_bc_seqba <- phyloseq::distance(physeq_M24_seqba, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_seqba <- data.frame(sample_data(physeq_M24_seqba))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_seqba~Sequencing_batch, data=metadata_physeq_M24_seqba) #ns
adonis(M24_unifrac_seqba~Sequencing_batch, data=metadata_physeq_M24_seqba) #sig
adonis(M24_jaccard_seqba~Sequencing_batch, data=metadata_physeq_M24_seqba) #sig
adonis(M24_bc_seqba~Sequencing_batch, data=metadata_physeq_M24_seqba) #sig
```
#### Duration of Any and Exclusive Breastfeeding
```{r, warning=FALSE}
#DURATION OF EXCLUSIVE BREASTFEEDING
physeq_M24_ebf <- subset_samples(physeq_M24, only_bf_months != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_ebf <- phyloseq::distance(physeq_M24_ebf, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_ebf <- phyloseq::distance(physeq_M24_ebf, method = "unifrac", type = "samples")
M24_jaccard_ebf <- phyloseq::distance(physeq_M24_ebf, method = "jaccard", type = "samples")
M24_bc_ebf <- phyloseq::distance(physeq_M24_ebf, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_ebf <- data.frame(sample_data(physeq_M24_ebf))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_ebf~only_bf_months, data=metadata_physeq_M24_ebf)#significant
adonis(M24_unifrac_ebf~only_bf_months, data=metadata_physeq_M24_ebf) #significant
adonis(M24_jaccard_ebf~only_bf_months, data=metadata_physeq_M24_ebf) #significant
adonis(M24_bc_ebf~only_bf_months, data=metadata_physeq_M24_ebf) #significant

#DURATION OF ANY BREASTFEEDING
physeq_M24_abf <- subset_samples(physeq_M24, any_bf_months != "NA")

#calculate distance matrices
set.seed(main.seed)  
M24_wunifrac_abf <- phyloseq::distance(physeq_M24_abf, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_abf <- phyloseq::distance(physeq_M24_abf, method = "unifrac", type = "samples")
M24_jaccard_abf <- phyloseq::distance(physeq_M24_abf, method = "jaccard", type = "samples")
M24_bc_abf <- phyloseq::distance(physeq_M24_abf, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_abf <- data.frame(sample_data(physeq_M24_abf))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_abf~any_bf_months, data=metadata_physeq_M24_abf) 
set.seed(main.seed)
adonis(M24_unifrac_abf~any_bf_months, data=metadata_physeq_M24_abf) 
adonis(M24_jaccard_abf~any_bf_months, data=metadata_physeq_M24_abf) 
adonis(M24_bc_abf~any_bf_months, data=metadata_physeq_M24_abf) 

#Determine if breastfeeding variables collinear; if so, only use one
alpha_diversity_stress_usable %$%
  ctable(only_bf_months, any_bf_months,
    prop = "r", chisq = TRUE, headings = FALSE
  ) %>%
  print(
    method = "render",
    style = "rmarkdown",
    footnote = NA
  )
```
#### Diet covariates

```{r}
#FIBER
physeq_M24_fiber <- subset_samples(physeq_M24, Fibre.per.1000kcal != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_fiber <- phyloseq::distance(physeq_M24_fiber, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_fiber <- phyloseq::distance(physeq_M24_fiber, method = "unifrac", type = "samples")
M24_jaccard_fiber <- phyloseq::distance(physeq_M24_fiber, method = "jaccard", type = "samples")
M24_bc_fiber <- phyloseq::distance(physeq_M24_fiber, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_fiber <- data.frame(sample_data(physeq_M24_fiber))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_fiber~Fibre.per.1000kcal, data=metadata_physeq_M24_fiber)#significant
set.seed(main.seed)
adonis(M24_unifrac_fiber~Fibre.per.1000kcal, data=metadata_physeq_M24_fiber) #n.s.
set.seed(main.seed)
adonis(M24_jaccard_fiber~Fibre.per.1000kcal, data=metadata_physeq_M24_fiber) #significant
set.seed(main.seed)
adonis(M24_bc_fiber~Fibre.per.1000kcal, data=metadata_physeq_M24_fiber) #significant

#PUFA
physeq_M24_pufa <- subset_samples(physeq_M24, PUFA_. != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_pufa <- phyloseq::distance(physeq_M24_pufa, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_pufa  <- phyloseq::distance(physeq_M24_pufa, method = "unifrac", type = "samples")
M24_jaccard_pufa  <- phyloseq::distance(physeq_M24_pufa, method = "jaccard", type = "samples")
M24_bc_pufa  <- phyloseq::distance(physeq_M24_pufa, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_pufa  <- data.frame(sample_data(physeq_M24_pufa))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_pufa~PUFA_., data=metadata_physeq_M24_pufa)#significant
set.seed(main.seed)
adonis(M24_unifrac_pufa~PUFA_., data=metadata_physeq_M24_pufa) #n.s.
set.seed(main.seed)
adonis(M24_jaccard_pufa~PUFA_., data=metadata_physeq_M24_pufa) #significant
set.seed(main.seed)
adonis(M24_bc_pufa~PUFA_., data=metadata_physeq_M24_pufa) #significant

#PROTEIN
physeq_M24_protein <- subset_samples(physeq_M24, `Protein_.energy` != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_protein <- phyloseq::distance(physeq_M24_protein, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_protein  <- phyloseq::distance(physeq_M24_protein, method = "unifrac", type = "samples")
M24_jaccard_protein  <- phyloseq::distance(physeq_M24_protein, method = "jaccard", type = "samples")
M24_bc_protein  <- phyloseq::distance(physeq_M24_protein, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_protein  <- data.frame(sample_data(physeq_M24_protein))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_protein~`Protein_.energy`, data=metadata_physeq_M24_protein)
set.seed(main.seed)
adonis(M24_unifrac_protein~`Protein_.energy`, data=metadata_physeq_M24_protein) 
set.seed(main.seed)
adonis(M24_jaccard_protein ~`Protein_.energy`, data=metadata_physeq_M24_protein) 
set.seed(main.seed)
adonis(M24_bc_protein~`Protein_.energy`, data=metadata_physeq_M24_protein) 

#FAT (overall)
physeq_M24_fat <- subset_samples(physeq_M24, `Fat_.energy` != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_fat <- phyloseq::distance(physeq_M24_fat, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_fat  <- phyloseq::distance(physeq_M24_fat, method = "unifrac", type = "samples")
M24_jaccard_fat  <- phyloseq::distance(physeq_M24_fat, method = "jaccard", type = "samples")
M24_bc_fat  <- phyloseq::distance(physeq_M24_fat, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_fat  <- data.frame(sample_data(physeq_M24_fat))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_fat~`Fat_.energy`, data=metadata_physeq_M24_fat)
set.seed(main.seed)
adonis(M24_unifrac_fat~`Fat_.energy`, data=metadata_physeq_M24_fat) 
set.seed(main.seed)
adonis(M24_jaccard_fat ~`Fat_.energy`, data=metadata_physeq_M24_fat) 
set.seed(main.seed)
adonis(M24_bc_fat~`Fat_.energy`, data=metadata_physeq_M24_fat) 


#SAT FAT
physeq_M24_satfat <- subset_samples(physeq_M24, SatFat_. != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_satfat <- phyloseq::distance(physeq_M24_satfat, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_satfat  <- phyloseq::distance(physeq_M24_satfat, method = "unifrac", type = "samples")
M24_jaccard_satfat  <- phyloseq::distance(physeq_M24_satfat, method = "jaccard", type = "samples")
M24_bc_satfat  <- phyloseq::distance(physeq_M24_satfat, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_satfat  <- data.frame(sample_data(physeq_M24_satfat))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_satfat~SatFat_., data=metadata_physeq_M24_satfat)
set.seed(main.seed)
adonis(M24_unifrac_satfat~SatFat_., data=metadata_physeq_M24_satfat) 
set.seed(main.seed)
adonis(M24_jaccard_satfat ~SatFat_., data=metadata_physeq_M24_satfat) 
set.seed(main.seed)
adonis(M24_bc_satfat~SatFat_., data=metadata_physeq_M24_satfat) 

#MUFA
physeq_M24_mufa <- subset_samples(physeq_M24,  MUFA_. != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_mufa <- phyloseq::distance(physeq_M24_mufa, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_mufa  <- phyloseq::distance(physeq_M24_mufa, method = "unifrac", type = "samples")
M24_jaccard_mufa  <- phyloseq::distance(physeq_M24_mufa, method = "jaccard", type = "samples")
M24_bc_mufa  <- phyloseq::distance(physeq_M24_mufa, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_mufa  <- data.frame(sample_data(physeq_M24_mufa))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_mufa~MUFA_., data=metadata_physeq_M24_mufa)
set.seed(main.seed)
adonis(M24_unifrac_mufa~MUFA_., data=metadata_physeq_M24_mufa) 
set.seed(main.seed)
adonis(M24_jaccard_mufa ~MUFA_., data=metadata_physeq_M24_mufa) 
set.seed(main.seed)
adonis(M24_bc_mufa~MUFA_., data=metadata_physeq_M24_mufa)

#CARBOHYDRATES
physeq_M24_carbs <- subset_samples(physeq_M24,  `CHO_.energy` != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_carbs <- phyloseq::distance(physeq_M24_carbs, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_carbs  <- phyloseq::distance(physeq_M24_carbs, method = "unifrac", type = "samples")
M24_jaccard_carbs  <- phyloseq::distance(physeq_M24_carbs, method = "jaccard", type = "samples")
M24_bc_carbs  <- phyloseq::distance(physeq_M24_carbs, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_carbs  <- data.frame(sample_data(physeq_M24_carbs))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_carbs~`CHO_.energy`, data=metadata_physeq_M24_carbs)
set.seed(main.seed)
adonis(M24_unifrac_carbs~`CHO_.energy`, data=metadata_physeq_M24_carbs) 
set.seed(main.seed)
adonis(M24_jaccard_carbs ~`CHO_.energy`, data=metadata_physeq_M24_carbs) 
set.seed(main.seed)
adonis(M24_bc_carbs ~`CHO_.energy`, data=metadata_physeq_M24_carbs)

#DQI
physeq_M24_dqi <- subset_samples(physeq_M24, DQI.score.adjusted != "NA")

#calculate distance matrices
set.seed(main.seed) 
M24_wunifrac_dqi <- phyloseq::distance(physeq_M24_dqi, method = "wunifrac", type = "samples")
set.seed(main.seed) 
M24_unifrac_dqi  <- phyloseq::distance(physeq_M24_dqi, method = "unifrac", type = "samples")
M24_jaccard_dqi  <- phyloseq::distance(physeq_M24_dqi, method = "jaccard", type = "samples")
M24_bc_dqi  <- phyloseq::distance(physeq_M24_dqi, method = "bray", type = "samples")

#get dataframe of metadata
metadata_physeq_M24_dqi  <- data.frame(sample_data(physeq_M24_dqi))

#perform PERMANOVAs with each distance matrix
set.seed(main.seed)
adonis(M24_wunifrac_dqi~DQI.score.adjusted, data=metadata_physeq_M24_dqi)
set.seed(main.seed)
adonis(M24_unifrac_dqi~DQI.score.adjusted, data=metadata_physeq_M24_dqi) 
set.seed(main.seed)
adonis(M24_jaccard_dqi ~DQI.score.adjusted, data=metadata_physeq_M24_dqi) 
set.seed(main.seed)
adonis(M24_bc_dqi ~DQI.score.adjusted, data=metadata_physeq_M24_dqi)
```

**Summary**

- *Significant covariates:* ethnicity, delivery mode, duration of exclusive breastfeeding, antibiotics, sequencing batch, fiber, PUFA   
- Even though both duration of exclusive and partial breastfeeding were significant for the sake of model parsimony only used duration of exclsuive breastfeeding because the two breastfeeing variables are associated with each other and duration of exclusive was sig for more beta diversity metrics

### All Adversity Exposures
#### Including Covariates
```{r, warning=FALSE}
#INCLUDING COVARIATES
#create a phyloseq object with complete cases on adversity variables and covaraites
physeq_M24_all <- subset_samples(physeq_M24, !is.na(postnatal_stress) & !is.na(maltreated_mod_rev) & !is.na(stai_state_above_clin_cutoff_pw26) & !is.na(only_bf_months) & !is.na(deliv_mode) & !is.na(mother_ethnicity) & !is.na(antibio_M24) & !is.na(Sequencing_batch) & !is.na(PUFA_.)) #N=157 for this analysis

#calculate distance matrices using this complete cases dataset
set.seed(main.seed)
M24_wunifrac_all <- phyloseq::distance(physeq_M24_all, method = "wunifrac", type = "samples")
set.seed(main.seed)
M24_unifrac_all <- phyloseq::distance(physeq_M24_all, method = "unifrac", type = "samples")
M24_bc_all <- phyloseq::distance(physeq_M24_all, method = "bray", type = "samples")
M24_jaccard_all <- phyloseq::distance(physeq_M24_all, method = "jaccard", type = "samples")

#get metadata
metadata_physeq_M24_all <- data.frame(sample_data(physeq_M24_all)) #get data frame of metadata

#test assumption of betadispersion
disp_all_maltreated <- betadisper(M24_wunifrac_all, metadata_physeq_M24_all$maltreated_mod_rev)
permutest(disp_all_maltreated) #yes, apparently they do have homogeneity of dispersion
disp_all_stai <- betadisper(M24_wunifrac_all, metadata_physeq_M24_all$stai_state_above_clin_cutoff_pw26)
permutest(disp_all_stai) #yes, apparently they do have homogeneity of dispersion
disp_all_leq <- betadisper(M24_wunifrac_all, metadata_physeq_M24_all$postnatal_stress)
permutest(disp_all_leq) #yes, apparently they do have homogeneity of dispersion

#permanovas
#weighted unifrac
set.seed(main.seed)
wunifrac_all_covs <- adonis(M24_wunifrac_all~ maltreated_mod_rev+stai_state_above_clin_cutoff_pw26+postnatal_stress +antibio_M24+deliv_mode+only_bf_months+mother_ethnicity+Fibre.per.1000kcal+PUFA_.+Sequencing_batch, data=metadata_physeq_M24_all) 
wunifrac_all_covs$aov.tab #print output
omega_sq_adonis(wunifrac_all_covs) #calculates omega squared for first predictor; change order of vars to get omega squared for other predictors

#unweighted unifrac
set.seed(main.seed)
unifrac_all_covs <- adonis(M24_unifrac_all~ maltreated_mod_rev +stai_state_above_clin_cutoff_pw26+postnatal_stress+antibio_M24+deliv_mode+only_bf_months+mother_ethnicity+Fibre.per.1000kcal+PUFA_.+Sequencing_batch, data=metadata_physeq_M24_all) 
unifrac_all_covs$aov.tab #print output
omega_sq_adonis(unifrac_all_covs)

#bray-curtis
bc_all_covs <- adonis(M24_bc_all~ maltreated_mod_rev+stai_state_above_clin_cutoff_pw26+postnatal_stress +antibio_M24+deliv_mode+only_bf_months+mother_ethnicity+Fibre.per.1000kcal+PUFA_.+Sequencing_batch, data=metadata_physeq_M24_all) 
bc_all_covs$aov.tab #print output
omega_sq_adonis(bc_all_covs)

#jaccard
jacc_all_covs <- adonis(M24_jaccard_all~maltreated_mod_rev +stai_state_above_clin_cutoff_pw26+postnatal_stress+antibio_M24+deliv_mode+only_bf_months+mother_ethnicity+Fibre.per.1000kcal+PUFA_.+Sequencing_batch, data=metadata_physeq_M24_all) 
jacc_all_covs$aov.tab #print output
omega_sq_adonis(jacc_all_covs)
```

#### Not Including Covariates
```{r, warning=FALSE}
#NOT INCLUDING COVARIATES
#using dataset with complete cases on all adversities and covariates

#calculate distance matrices
set.seed(main.seed)
M24_wunifrac_all_ncovs <- phyloseq::distance(physeq_M24_all, method = "wunifrac", type = "samples")
set.seed(main.seed)
M24_unifrac_all_ncovs <- phyloseq::distance(physeq_M24_all, method = "unifrac", type = "samples")
M24_bc_all_ncovs <- phyloseq::distance(physeq_M24_all, method = "bray", type = "samples")
M24_jaccard_all_ncovs <- phyloseq::distance(physeq_M24_all, method = "jaccard", type = "samples")

#get metadata
metadata_physeq_M24_all_ncovs <- data.frame(sample_data(physeq_M24_all)) #get data frame of metadata

#permanovas
#weighted unifrac
set.seed(main.seed)
wunifrac_all_ncovs <- adonis(M24_wunifrac_all_ncovs~ stai_state_above_clin_cutoff_pw26+maltreated_mod_rev+postnatal_stress, data=metadata_physeq_M24_all_ncovs)
wunifrac_all_ncovs$aov.tab #print output
omega_sq_adonis(wunifrac_all_ncovs)

#unweighted unifrac
set.seed(main.seed)
unifrac_all_ncovs <- adonis(M24_unifrac_all_ncovs~stai_state_above_clin_cutoff_pw26+postnatal_stress+maltreated_mod_rev, data=metadata_physeq_M24_all_ncovs)
unifrac_all_ncovs$aov.tab
omega_sq_adonis(unifrac_all_ncovs)

#bray-curtis
bc_all_ncovs <- adonis(M24_bc_all_ncovs~postnatal_stress+maltreated_mod_rev+stai_state_above_clin_cutoff_pw26, data=metadata_physeq_M24_all_ncovs)
bc_all_ncovs$aov.tab
omega_sq_adonis(bc_all_ncovs)

#jaccard
jacc_all_ncovs <- adonis(M24_jaccard_all_ncovs~ postnatal_stress+maltreated_mod_rev+stai_state_above_clin_cutoff_pw26, data=metadata_physeq_M24_all_ncovs)
jacc_all_ncovs$aov.tab
omega_sq_adonis(jacc_all_ncovs)
```

### Cumulative Adversity
#### Including Covariates
```{r, warning=FALSE}
#get dataset
physeq_M24_cumulative <- subset_samples(physeq_M24, !is.na(cumulative_adv) & !is.na(only_bf_months) & !is.na(deliv_mode) & !is.na(mother_ethnicity) & !is.na(antibio_M24) & !is.na(Sequencing_batch) & !is.na(PUFA_.)) #N=157

#calculate distance matrices
set.seed(main.seed)
M24_wunifrac_cumulative <- phyloseq::distance(physeq_M24_cumulative, method = "wunifrac", type = "samples")
set.seed(main.seed)
M24_unifrac_cumulative <- phyloseq::distance(physeq_M24_cumulative, method = "unifrac", type = "samples")
M24_bc_cumulative <- phyloseq::distance(physeq_M24_cumulative, method = "bray", type = "samples")
M24_jaccard_cumulative <- phyloseq::distance(physeq_M24_cumulative, method = "jaccard", type = "samples")

#get metadata
metadata_physeq_M24_cumulative <- data.frame(sample_data(physeq_M24_cumulative)) #get data frame of metadata

#test assumption of betadispersion
disp_cumulative<- betadisper(M24_wunifrac_cumulative, metadata_physeq_M24_cumulative$cumulative_adv)
permutest(disp_cumulative) #yes, apparently they do have homogeneity of dispersion
plot(disp_cumulative, hull=FALSE, ellipse=TRUE)

#permanovas 
#weighted unifrac
set.seed(main.seed)
wunifrac_cumulative_covs <- adonis(M24_wunifrac_cumulative~cumulative_adv+antibio_M24+mother_ethnicity+deliv_mode+only_bf_months+Fibre.per.1000kcal+PUFA_.+Sequencing_batch, data=metadata_physeq_M24_cumulative) 
wunifrac_cumulative_covs
omega_sq_adonis(wunifrac_cumulative_covs)

#unweighted unifrac
set.seed(main.seed)
unifrac_cumulative_covs <- adonis(M24_unifrac_cumulative~cumulative_adv+antibio_M24+mother_ethnicity+deliv_mode+only_bf_months+Fibre.per.1000kcal+PUFA_.+Sequencing_batch, data=metadata_physeq_M24_cumulative)
unifrac_cumulative_covs
omega_sq_adonis(unifrac_cumulative_covs)

#bray-curtis
bc_cumulative_covs <- adonis(M24_bc_cumulative~cumulative_adv+antibio_M24+mother_ethnicity+deliv_mode+only_bf_months+Fibre.per.1000kcal+PUFA_.+Sequencing_batch, data=metadata_physeq_M24_cumulative)
bc_cumulative_covs
omega_sq_adonis(bc_cumulative_covs)

#jaccard
jacc_cumulative_covs <- adonis(M24_jaccard_cumulative~cumulative_adv+antibio_M24+mother_ethnicity+deliv_mode+only_bf_months+Fibre.per.1000kcal+PUFA_.+Sequencing_batch, data=metadata_physeq_M24_cumulative)
jacc_cumulative_covs
omega_sq_adonis(jacc_cumulative_covs)
```

#### Not Including Covariates
```{r, warning=FALSE}
#using same dataset as above, physeq_M24_cumulative

#calculate distance matrices
set.seed(main.seed)
M24_wunifrac_cumulative_ncovs <- phyloseq::distance(physeq_M24_cumulative, method = "wunifrac", type = "samples")
set.seed(main.seed)
M24_unifrac_cumulative_ncovs <- phyloseq::distance(physeq_M24_cumulative, method = "unifrac", type = "samples")
M24_bc_cumulative_ncovs <- phyloseq::distance(physeq_M24_cumulative, method = "bray", type = "samples")
M24_jaccard_cumulative_ncovs <- phyloseq::distance(physeq_M24_cumulative, method = "jaccard", type = "samples")

#get metadata
metadata_physeq_M24_cumulative_nocovs <- data.frame(sample_data(physeq_M24_cumulative)) #get data frame of metadata

#PERMANOVAS
#weighted unifrac
set.seed(main.seed)
wunifrac_cumulative_ncovs <- adonis(M24_wunifrac_cumulative_ncovs~cumulative_adv, data=metadata_physeq_M24_cumulative_nocovs)
wunifrac_cumulative_ncovs$aov.tab
omega_sq_adonis(wunifrac_cumulative_ncovs)

#unweighted unifrac
set.seed(main.seed)
unifrac_cumulative_ncovs <- adonis(M24_unifrac_cumulative_ncovs~cumulative_adv, data=metadata_physeq_M24_cumulative_nocovs)
unifrac_cumulative_ncovs$aov.tab
omega_sq_adonis(unifrac_cumulative_ncovs)

#bray-curtis
bc_cumulative_ncovs <- adonis(M24_bc_cumulative_ncovs~cumulative_adv, data=metadata_physeq_M24_cumulative_nocovs)
bc_cumulative_ncovs$aov.tab
omega_sq_adonis(bc_cumulative_ncovs)

#jaccard
jacc_cumulative_ncovs <- adonis(M24_jaccard_cumulative_ncovs~cumulative_adv, data=metadata_physeq_M24_cumulative_nocovs)
jacc_cumulative_ncovs$aov.tab
omega_sq_adonis(jacc_cumulative_ncovs)
```

## Differential Abundance (DA) 
**Filtering out low prevalence taxa**:     
In the literature, different filtering thresholds are used based on data features and research questions (e.g., whether rare taxa are considered important) ([Carlson et al. (2021)](https://www.nature.com/articles/s41467-021-23281-y) Nature Communications).   
For these analyses we chose to report the results after filtering at a moderate threshold 0.5%, and included a more conservative (0.1%) and more liberal (1%) threshold in the supplemental material:  

* at least 1% of samples have non-zero abundance (keeps 650 out of 1230 taxa; reported in supplement)
* at least 0.5% of samples have non-zero abundance (keeps 773 out of 1230 taxa; reported in main text)
* at least 0.1%of samples have non-zero abundance (keeps 1055 out of 1230 taxa, only removing taxa that are zero for all samples; reported in supplement)  

### Format data for DA analyses with MaAsLin2
Maaslin2 requires as input:  
1. A tab-delimited metadata file (features as columns, samples as rows)  
2. A tab-delimited feature file (taxonomy as columns, samples as rows)  
```{r eval=FALSE}
#transform OTU counts to relative abundances
physeq_M24_relabund = transform_sample_counts(physeq_M24, function(x) x / sum(x))

#convert phyloseq OTU table to dataframe
M24_otus = as(otu_table(physeq_M24_relabund), "matrix") #convert from part of phyloseq object to matrix
M24_otus_df = as.data.frame(M24_otus) #convert to df

#provide name for first column (for merging)
M24_ra_otus_df <-
  M24_otus_df %>% 
  rownames_to_column(var = "Feature.ID")

#join taxonomy file to OTU table to get taxa names for OTUs
taxonomy_txt_path <- str_c(general_mb_path, "input_files/M2448_taxonomy_edited.tsv") #taxonomy file path
M2448_tax <- read.table(taxonomy_txt_path, header=TRUE, sep = "\t") #read in taxa

M24_ra_tax_features <-
  M2448_tax %>% 
  full_join(M24_ra_otus_df, by = "Feature.ID")

M24_ra_tax_features <-
  M24_ra_tax_features %>% 
  filter('Feature.ID' != "OTU948") #filter out one OTU/taxa which has NA values for all samples (filtered out for alpha diversity calculations in qiime2 preprocessing)

#remove OTU names
M24_ra_tax_features <-
  M24_ra_tax_features %>% 
  dplyr::select(
    -'Feature.ID'
  )

#transpose this taxonomy feature file so that taxonomy are columns
M24_ra_tax_features_t <- t(M24_ra_tax_features)
M24_ra_tax_features_t = setNames(data.frame(t(M24_ra_tax_features[,-1])), M24_ra_tax_features[,1])

#name sample ID column for matching with metadata in maaslin2
M24_ra_tax_features_t <-
  M24_ra_tax_features_t %>% 
  rownames_to_column(var = '#SampleID') 

#write new file combining taxonomy and OTU counts as .tsv
tax_features_ra_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/M24_tax_features_ra.tsv"
write_tsv(M24_ra_tax_features_t, tax_features_ra_fp)
#also write these data to form B data folder for submission with GUSTO form B application
write_csv(M24_ra_tax_features_t, "../../manuscripts/fran_adversity_microbiome/Form_B_submission/data_updated/microbiome_taxonomy_and_counts.csv")

#read in metadata as dataframe
M24_metadata <- read_tsv(str_c(general_mb_path, "input_files/M24_metadata_typeheader_new_rev.tsv"))

#remove typeheader so that variable type can be read correctly by R
M24_metadata <- M24_metadata[-1, ]

#write metadata file with M24 samples as .tsv
M24_meta_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/M24_metadata.tsv"
write_tsv(M24_metadata, M24_meta_fp)
#manually open the .tsv and save it as .csv
csv_file <- "../../data/analysis_datasets/fran/maaslin2/inputs/M24_metadata.csv"
#read .csv into R -- after removing the typeheader, this will automatically assign each variable to the correct type
test <- read_csv(csv_file)

#manually assign correct variable type to binary variables that were read as double (numeric with decimal place) so that maaslin2 will treat those variables as categorical
test <- test %>% 
    dplyr::mutate(
      across(.cols = c(maltreated_mod_rev, stai_state_above_clin_cutoff_pw26, antibio_M24, probiotics_M24, cumulative_adv), ~as.character(.x)),
      across(.cols = c(only_bf_months, mother_ethnicity), ~as.factor(.x)))

#create new descriptive variables for easier visualization when graphing
test <- test %>% 
  mutate(sex_binary_ch = case_when( 
    sex_binary == 1 ~ "male",
    sex_binary== 0 ~ "female"
  ),
  postnatal_stress_s = case_when(
    postnatal_stress==1 ~ "yes",
    postnatal_stress==0 ~ "no"
  ), 
  cumulative_adv_s = case_when(
    cumulative_adv == 0 ~ "0 tp",
    cumulative_adv == 1 ~ "1 tp",
    cumulative_adv == 2 ~ "2 tp",
    cumulative_adv == 3 ~ "3+ tp"
  )
  )

#remove variables that won't be used in maaslin2 analyses
test <- test %>% 
  dplyr::select(
    -contains("mom"),
    -sum_advrsty,
    -deliv_mode
  )

#write new metadata file with correct dadta types back to original file path
write_tsv(test, "../../data/analysis_datasets/fran/maaslin2/inputs/M24_metadata.tsv")

#load final metadata file for analysis below
M24_metadata <- read_tsv("../../data/analysis_datasets/fran/maaslin2/inputs/M24_metadata.tsv")
```
### Covariate Selection
**Covariates will be chosen based on overlap in significantly different bacteria with adversity maaslin results without covariates at the genus level.**   

Possible covariates include: sequencing batch, antibiotics, probiotics, sex, delivery mode, ethnicity, duration of exclusive breastfeeding, income, gestational age at birth, age at stool sample proxy, diet covariates.      

Same Maaslin2 default parameters as used above and 0.5% filtering threshold.   

#### Child Sex  
```{r eval=FALSE}
#child sex
maaslin2_sex_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/sex/sex_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("sex_binary_ch"),
  standardize = TRUE 
)
```
#### Antibiotic Use
```{r eval=FALSE}
#antibiotic use
maaslin2_antibiotics_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/antibiotics/antibiotiics_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("antibio_M24"),
  standardize = TRUE 
)
```
#### Probiotic Use
```{r eval=FALSE}
#probiotic use
maaslin2_probiotics_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/probiotics/probiotics_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("probiotics_M24"),
  standardize = TRUE
)
```
#### Delivery Mode
```{r eval=FALSE}
#delivery mode
maaslin2_delivmode_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/delivery_mode/delivmode_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("deliv_mode_str"),
  standardize = TRUE 
)
```
#### Child Ethnicity
```{r eval=FALSE}
#child ethnicity
maaslin2_ethnicity_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/ethnicity/ethnicity_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)' 
)
```
#### Age at Stool Sample
```{r eval=FALSE}
#age at stool sample
maaslin2_age_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/age/age_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("c_age_days_stoolproxy_m24"),
  standardize = TRUE
)
```
#### Income
```{r eval=FALSE}
#monthly household income per member
maaslin2_income_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/income/income_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25,
  analysis_method = "lm", 
  fixed_effects = c("monthly_income_per_member"),
  standardize = TRUE
)
```
#### Gestational Age at Birth
```{r eval=FALSE}
#gestational age at birth
maaslin2_ga_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/gestational_age/ga_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("GA"),
  standardize = TRUE
)
```
#### Duration of Exclusive Breastfeeding
```{r eval=FALSE}
#duration of exclusive breastfeeding
maaslin2_exclusivebf_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/duration_of_exclusive_bf/exclusivebf_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("only_bf_months"),
  standardize = TRUE,
  reference = 'only_bf_months, c(0, 1, 2, 4)' #provide reference level for variables with more than 2 categories
)
```
#### Duration of Any Breastfeeding
```{r eval=FALSE}
#duration of any breastfeeding
maaslin2_anybf_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/duration_of_any_bf/anybf_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("any_bf_months"),
  standardize = TRUE, 
  reference = 'any_bf_months, c(0, 1, 2, 3)'
)
```
#### Sequencing Batch
```{r eval=FALSE}
#sequencing batch
maaslin2_seqbatch_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/sequencing_batch/seqbatch_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("Sequencing_batch"),
  standardize = TRUE 
)
```
#### Fiber
```{r}
maaslin2_fiber_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/fiber/fiber_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("Fibre.per.1000kcal"),
  standardize = TRUE 
)

#count the number of people with fiber data out of alpha_div_stress_usable (will use cutoff for sig results to examine at 15% of that)
alpha_diversity_stress_usable %>% dplyr::count(!is.na(Fibre.per.1000kcal)) #N=298 with Fiber info; .15 x 298 = 45 people minimum 
```
#### PUFA
```{r}
maaslin2_pufa_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/pufa/pufa_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("PUFA_."),
  standardize = TRUE 
)

#count the number of people with PUFA data out of alpha_div_stress_usable (will use cutoff for sig results to examine at 15% of that)
alpha_diversity_stress_usable %>% dplyr::count(!is.na(PUFA_.)) #N=298 with PUFA info; .15 x 298 = 45 people minimum 
```
#### MUFA
```{r}
maaslin2_mufa_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/mufa/mufa_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("MUFA_."),
  standardize = TRUE 
)
```
#### SatFat
```{r}
maaslin2_satfat_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/satfat/satfat_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("SatFat_."),
  standardize = TRUE 
)
```
#### Carbs
```{r}
maaslin2_carbs_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/carbs/carbs_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("`CHO_.energy`"),
  standardize = TRUE 
)
```
#### Protein
```{r}
maaslin2_protein_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/protein/protein_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("`Protein_.energy`"),
  standardize = TRUE 
)
```

#### DQI
```{r}
maaslin2_dqi_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/only_covs/dqi/dqi_0.5p_results",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("DQI.score.adjusted"),
  standardize = TRUE 
)
```


**Significant covariates for maaslin2 adversity analyses**:   

* Preconception adversity: Duration of exclusive breastfeeding, child ethnicity, sequencing batch, Fiber (clostridium sensu stricto overlap)  
* Prenatal adversity: Duration of exclusive breastfeeding, probiotic use, child ethnicity, sequencing batch, Fiber (Lachnoclostridium overlap)  
* Postnatal adversity: Child ethnicity  
* All adversities together: Duration of exclusive breastfeeding, child ethnicity, probiotic use, sequencing batch, and child age at stool sample  
* Cumulative adversity: none  

## Without Covariates
### All Adversity Exposures
**Note:** to perform the MaAsLin2 analyses, both an OTU file with all of the microbiome data, and a metadata file with all of the variables that will be associated with the microbiome data are needed. At the time of running these analyses, MaAsLin2 calculated the sample size, and the sample size with non-zero taxa abundance based on the OTU/taxonomy file. However, when a sample was missing some metadata, that sample would not be included in the analysis with the metadata variable (due to listwise deletion) and thus, the sample sizes would differ from that of the OTU file. This issue was brought to the attention of the [developers](https://forum.biobakery.org/t/maaslin2-n-and-n-not-0-output-appears-incorrect/3704). A simple fix to this problem would be to run MaAsLin2 on datasets with complete cases, to obtain the correct sample size. However, as we wanted to filter the data for non-zero abundance based on all samples with usable gut microbiome data, instead of subsets of the dataset based on metadata availability, we took the approach of running MaAsLin2 with all the cases included (in which instance, the sample sizes reported are incorrect) and again with complete cases for each metadata variable (in which instance the filtering for non zero abundance and thus, the results are incorrect, but the sample sizes are correct)

To reduce space, instead of writing out each version separately below, we ran each code chunk twice, swapping out the metadata file for all vs. complete cases and changing the folder name for the results (adding "_allcases" and "_completecases" for 1 and 2 respectively). Analysis objects below are saved using the all cases version for use in visualizations later, as that is the version with the coefficient estimates that we used.   
```{r eval=FALSE}
#ALL ADVERSITIES TOGETHER
#use same dataset as for analysis with covariates to ensure you're not just measuring a different sample
alladv_metadata <- M24_metadata %>% 
  filter(!is.na(postnatal_stress) & !is.na(STAI_prorated_state_pw26) & !is.na(ctq_total_score_25it) & !is.na(mother_ethnicity) & !is.na(only_bf_months) & !is.na(probiotics_M24) & !is.na(Sequencing_batch) & !is.na(c_age_years_stoolproxy_m24)) #N=196
write_tsv(alladv_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_alladv.tsv")
alladv_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_alladv.tsv"

#All adversities -- 0.5% filtering
maaslin2_alladv_nocov_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = alladv_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/no_covs/all_adversities/alladv_nocovs_0.5p_results_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("postnatal_stress", "STAI_prorated_state_pw26", "ctq_total_score_25it"),
  standardize = TRUE 
)
```

### Cumulative Adversity
```{r eval=FALSE}
#CUMULATIVE ADVERSITY
#create complete cases dataset to determine N and N.not.0
cumulative_metadata_nocovs <- M24_metadata %>% 
  filter(!is.na(cumulative_adv_s)) 
write_tsv(cumulative_metadata_nocovs, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_cumulative_nocovs.tsv")
cumulative_metadata_nocovs_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_cumulative_nocovs.tsv"

#Cumulative adversity -- 1% filtering
maaslin2_cumulative_nocov_1 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/no_covs/cumulative_adversity/cumulative_nocovs_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("cumulative_adv_s"),
  standardize = TRUE, 
  reference = c("cumulative_adv_s,0 tp") #provide reference level for variables with more than 2 categories
)

#Cumulative adversity -- 0.5% filtering
maaslin2_cumulative_nocov_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/no_covs/cumulative_adversity/cumulative_nocovs_0.5p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm",
  fixed_effects = c("cumulative_adv_s"),
  standardize = TRUE, 
  reference = c("cumulative_adv_s,0 tp") 
)

#Cumulative adversity -- 0.1% filtering
maaslin2_cumulative_nocov_0.1 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/no_covs/cumulative_adversity/cumulative_nocovs_0.1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cumulative_adv_s"),
  standardize = TRUE, 
  reference = c("cumulative_adv_s,0 tp")
)
```

### Each Adversity Exposure
#### Preconception Adversity
```{r eval=FALSE}
#PRECONCEPTION ADVERSITY
#create dataset with complete cases for cov analysis so that it is the same sample
precon_metadata <- M24_metadata %>% 
  dplyr::filter(!is.na(ctq_total_score_25it) & !is.na(Sequencing_batch) & !is.na(only_bf_months) & !is.na(mother_ethnicity) & !is.na(Fibre.per.1000kcal)) 
write_tsv(precon_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_precon.tsv")
precon_meta_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_precon.tsv"
  
#preconception adversity -- 0.5% filtering
maaslin_mothermalt_nocov_05 <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = precon_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/no_covs/mother_maltreatment/maltreated_nocovs_0.5p_results_covsample",
  min_abundance = 0,
  min_prevalence = 0.005, #0.5% filtering threshold
  max_significance = 0.25, #q-value threshold recommended for biomarker discovery
  #defaults: normalization = TSS and transform = log
  analysis_method = "lm", #default
  fixed_effects = c("ctq_total_score_25it"),
  standardize = TRUE #this option only affects the analysis when there is greater than one predictor
)
```

#### Prenatal Adversity
```{r eval=FALSE}
#PRENATAL ADVERSITY
#create subsample that is equivalen to complete cases for covariate analysis
prenat_metadata <- M24_metadata %>% 
  filter(!is.na(STAI_prorated_state_pw26) & !is.na(Sequencing_batch) & !is.na(only_bf_months) & !is.na(probiotics_M24) & !is.na(mother_ethnicity) & !is.na(Fibre.per.1000kcal)) #N=420
write_tsv(prenat_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_prenat.tsv")
prenat_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_prenat.tsv"

#Prenatal adversity -- 0.5% filtering
maaslin_prenatalanx_nocov_05 <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = prenat_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/no_covs/prenatal_anxiety/prenatalanx_nocovs_0.5p_results_covsample",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25,
  analysis_method = "lm",
  fixed_effects = c("STAI_prorated_state_pw26"),
  standardize = TRUE 
)
```

#### Postnatal Adversity
```{r eval=FALSE}
#POSTNATAL ADVERSITY
postnat_metadata <- M24_metadata %>% 
  filter(!is.na(postnatal_stress) & !is.na(mother_ethnicity)) #N=310
write_tsv(postnat_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_postnat.tsv")
postnat_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_postnat.tsv"

#postnatal adversity - 0.5% filtering
maaslin2_postnatalstress_nocov_05 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = postnat_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/no_covs/postnatal_stress/poststress_nocovs_0.5p_results_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("postnatal_stress"),
  standardize = TRUE 
)
```

## Including Covariates

### All Adversity Exposures
The same process as described in the "without covariates" section is followed, in which analyses are run once with all cases (to get coefficient estimates) and once with complete cases (to get correct N and N.not.0). The same Maaslin2 default parameters as used above, and 0.1%, 0.5%, and 1% filtering thresholds are used. 
```{r eval=FALSE}
#ALL ADVERSITIES TOGETHER
#create all adversity metadata
alladv_metadata <- M24_metadata %>% 
  filter(!is.na(postnatal_stress) & !is.na(STAI_prorated_state_pw26) & !is.na(ctq_total_score_25it) & !is.na(mother_ethnicity) & !is.na(only_bf_months) & !is.na(probiotics_M24) & !is.na(Sequencing_batch) & !is.na(c_age_years_stoolproxy_m24)) #N=196

write_tsv(alladv_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_alladv.tsv")
alladv_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_alladv.tsv"

#All adversities -- 1% filtering threshold
maaslin2_alladv_withcov_1 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/all_adversities/alladv_wcovs_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("postnatal_stress_s", "STAI_prorated_state_pw26", "ctq_total_score_25it", "mother_ethnicity", "only_bf_months", "c_age_years_stoolproxy_m24", "Sequencing_batch"),
  standardize = TRUE, 
  reference = c("mother_ethnicity,chinese", "only_bf_months,<1M") #formatting: no spaces can exist between comma and level
)

#All adversities -- 0.5% filtering threshold
maaslin2_alladv_withcov_05 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/all_adversities/alladv_wcovs_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25,
  analysis_method = "lm", 
  fixed_effects = c("postnatal_stress_s", "STAI_prorated_state_pw26", "ctq_total_score_25it", "mother_ethnicity", "only_bf_months", "c_age_years_stoolproxy_m24", "Sequencing_batch"),
  standardize = TRUE,
  reference = c('mother_ethnicity,chinese', 'only_bf_months,<1M') 
)

#All adversities -- 0.1% filtering threshold
maaslin2_alladv_withcov_01 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = alladv_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/all_adversities/alladv_wcovs_0.1p_results_completecases_test",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm",
  fixed_effects = c("postnatal_stress_s", "STAI_prorated_state_pw26", "ctq_total_score_25it", "mother_ethnicity", "only_bf_months", "c_age_years_stoolproxy_m24", "Sequencing_batch"),
  standardize = TRUE, 
  reference = c('mother_ethnicity, c(0, 1, 2)', 'only_bf_months, c(3, 4, 5)') 
)

```

### Cumulative Adversity
There were no selected covariates for cumulative adversity, so we only ran analyses for cumulative adversity not including covariates.

### Each Adversity Exposure
#### Preconception Adversity
```{r eval=FALSE}
#PRECONCEPTION ADVERSITY
#create complete metadata files to be used for these
precon_metadata <- M24_metadata %>% 
  filter(!is.na(ctq_total_score_25it) & !is.na(Sequencing_batch) & !is.na(only_bf_months) & !is.na(mother_ethnicity) & !is.na(Fibre.per.1000kcal)) 
write_tsv(precon_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_precon.tsv")
precon_meta_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_precon.tsv"

#preconception adversity - 1% filtering threshold
maaslin_maltreatment_covs_1p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = precon_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/mother_maltreatment/mothermalt_wcovs_1p_results_compcases_fiber",
  min_abundance = 0,
  min_prevalence = 0.01, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "ctq_total_score_25it", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, #because we have multiple continuous metadata variables
  reference = c('mother_ethnicity,chinese', 'only_bf_months,<1M') 
)

#preconception adversity -- 0.5% filtering threshold
maaslin_maltreatment_covs_0.5p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/mother_maltreatment/mothermalt_wcovs_0.5p_results_allcases_fiber_test",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "ctq_total_score_25it", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity,chinese', 'only_bf_months,<1M') 
)

#preconception adversity -- 0.1% filtering threshold
maaslin_maltreatment_covs_0.1p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = precon_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/mother_maltreatment/mothermalt_wcovs_0.1p_results_compcases_fiber",
  min_abundance = 0,
  min_prevalence = 0.001, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "ctq_total_score_25it", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity,chinese', 'only_bf_months,<1M') 
)
```

#### Prenatal Adversity
```{r eval=FALSE}
#PRENATAL ADVERSITY
#create prenatal adversity metadata
prenat_metadata <- M24_metadata %>% 
  filter(!is.na(STAI_prorated_state_pw26) & !is.na(Sequencing_batch) & !is.na(only_bf_months) & !is.na(probiotics_M24) & !is.na(mother_ethnicity) & !is.na(Fibre.per.1000kcal)) #N=420
write_tsv(prenat_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_prenat.tsv")
prenat_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_prenat.tsv"


#prenatal adversity -- 1% filtering threshold 
maaslin_prenatanx_covs_1p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = prenat_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/prenatal_anxiety/prenatanx_wcovs_1p_results_compcases_fiber",
  min_abundance = 0,
  min_prevalence = 0.01, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "probiotics_M24", "STAI_prorated_state_pw26", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity, c(0, 1, 2)', 'only_bf_months, c(3, 4, 5)') 
)

#prenatal adversity -- 0.5% filtering threshold 
maaslin_prenatanx_covs_0.5p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/prenatal_anxiety/prenatanx_wcovs_0.5p_results_allcases_fiber",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("mother_ethnicity", "only_bf_months", "probiotics_M24", "STAI_prorated_state_pw26", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity, c(0, 1, 2)', 'only_bf_months, c(3, 4, 5)') 
)

#prenatal adversity -- 0.1% filtering threshold 
maaslin_prenatanx_covs_0.1p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = precon_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/prenatal_anxiety/prenatanx_wcovs_0.1p_results_compcases_fiber",
  min_abundance = 0,
  min_prevalence = 0.001, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "probiotics_M24", "STAI_prorated_state_pw26", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity, c(0, 1, 2)', 'only_bf_months, c(3, 4, 5)') 
)
```

#### Postnatal Adversity
```{r eval=FALSE}
#POSTNATAL ADVERSITY
#create postnatal adversity metadata
postnat_metadata <- M24_metadata %>% 
  filter(!is.na(postnatal_stress) & !is.na(mother_ethnicity)) #N=310
write_tsv(postnat_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_postnat.tsv")
postnat_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_postnat.tsv"

#postnatal adversity -- 1% filtering threshold
maaslin_poststress_covs_1p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/postnatal_stress/poststress_wcovs_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm",
  fixed_effects = c("mother_ethnicity", "postnatal_stress_s"),
  standardize = TRUE,
  reference = 'mother_ethnicity, c(0, 1, 2)'
)

#Postnatal adversity -- 0.05% filtering threshold
maaslin_poststress_covs_0.5p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/postnatal_stress/poststress_wcovs_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "postnatal_stress_s"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)' 
)

#Postnatal adversity -- 0.01% filtering threshold
maaslin_poststress_covs_0.1p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/postnatal_stress/poststress_wcovs_0.1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.001, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "postnatal_stress"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)' 
)
```


# Microbiome associations with child socioemotional functioning (SEF)
## Determine which SEF domains are correlated with adversities
*Note:* within subsample of participants that have microbiome data
```{r eval=FALSE}
#select adversity variables from usable dataset
cbcl_adversity <- alpha_diversity_stress_usable %>% 
  dplyr::select(
    subjid,
    ctq_total_score_25it,
    STAI_prorated_state_pw26,
    postnatal_stress,
    cumulative_adv
  )

#select all CBCL variables of interest from Y2 and Y4 score datasets
cbcl_Y4 <- read_csv(beh4_file) %>% 
  dplyr::select(
    subjid,
    contains("std")
  )

cbcl_Y2 <- read_csv(beh2_file) %>% 
  dplyr::select(
    subjid,
    contains("std")
  )

#look at correlations between cbcl subscales
cbcl_nosub <-
  child_beh %>%
  dplyr::select(-subjid)

corPlot(cbcl_nosub,
        numbers = TRUE,
        stars = TRUE,
        show.legend = TRUE,
        cex.axis = 0.8,
        cex = 0.5) 

#join cbcl into adversity variables - separately for each timepoint
cbcl2_adversity <- cbcl_adversity %>% 
  left_join( #note: doing left join bc we only care about cbcl within mb subsample
    cbcl_Y2, by = "subjid"
  ) 

cbcl4_adversity <- cbcl_adversity %>% 
  left_join(
    cbcl_Y4, by = "subjid"
  )

#create datasets with just continuous vars
cbcl2_adversity_cor <- cbcl2_adversity %>% 
  dplyr::select(
    -subjid,
    -postnatal_stress,
    -cumulative_adv
  )

cbcl4_adversity_cor <- cbcl4_adversity %>% 
  dplyr::select(
    -subjid,
    -postnatal_stress,
    -cumulative_adv
  )

#visualize cbcl year 2 correlations
corPlot(cbcl2_adversity_cor,
        numbers=TRUE,
        main = "",
        stars=TRUE,
        show.legend=TRUE,
        diag=FALSE,
        upper=FALSE,
        scale = TRUE,
        xlas=2,
        ylas=2,
#        gr=gr,
        cex.axis = 0.8,
        cex = 0.5)

#determine correlation values and which are significant after multiple comparison correction
cbcl2_cors <- corr.test(cbcl2_adversity_cor, method="pearson", use = "pairwise", adjust="BH")
View(cbcl2_cors$ci2) #view adjusted p-values. Note that these don't show the labels for which correlation it is. So determine the 'adjusted' p-value cutoff w/respect to the unadjusted p-values, and then go off of that on the df$ci dataframe

cbcl4_cors <- corr.test(cbcl4_adversity_cor, method="pearson", use = "pairwise", adjust="BH")
View(cbcl4_cors$ci2) #view adjusted pvalues

#visualize cbcl year 4 correlations
corPlot(cbcl4_adversity_cor,
        numbers=TRUE,
        main = "",
        stars=TRUE,
        show.legend=TRUE,
        diag=FALSE,
        upper=FALSE,
        scale = TRUE,
        xlas=2,
        ylas=2,
#        gr=gr,
        cex.axis = 0.8,
        cex = 0.5)

#t-tests for postnatal adversity and cbcl at age 2 years
#grab only needed variables
cbcl2_postnatal <- cbcl2_adversity %>% 
  dplyr::select(
    postnatal_stress,
    STAI_prorated_state_pw26,
    ctq_total_score_25it,
    contains("cbcl")
  )
#put data in long format to prep
cbcl2_postnatal.longer <- cbcl2_postnatal %>%
  pivot_longer(-postnatal_stress, names_to = "variables", values_to = "value")
cbcl2_postnatal.longer %>% sample_n(6)

#run t-tests
stat.test <- cbcl2_postnatal.longer %>%
  group_by(variables) %>%
  t_test(value ~ postnatal_stress) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance()
stat.test

#t-tests for postnatal adversity and cbcl outcomes year 4
#grab only needed variables
cbcl4_postnatal <- cbcl4_adversity %>% 
  dplyr::select(
    postnatal_stress,
    contains("cbcl")
  )
#put data in long format to prep
cbcl4_postnatal.longer <- cbcl4_postnatal %>%
  pivot_longer(-postnatal_stress, names_to = "variables", values_to = "value")
cbcl4_postnatal.longer %>% sample_n(6)

#run t-tests
stat.test <- cbcl4_postnatal.longer %>%
  group_by(variables) %>%
  t_test(value ~ postnatal_stress) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance()
stat.test


#anovas for cumulative adversity and cbcl outcomes at year 2 and  year 4
cbcl2_cumulative <- cbcl2_adversity %>% #grab only needed variables
  dplyr::select(
    cumulative_adv,
    contains("cbcl")
  )

cbcl4_cumulative <- cbcl4_adversity %>% #grab only needed variables
  dplyr::select(
    cumulative_adv,
    contains("cbcl")
  )

#put data in long format to prep
cbcl2_cumulative.longer <- cbcl2_cumulative %>%
  pivot_longer(-cumulative_adv, names_to = "variables", values_to = "value")

cbcl4_cumulative.longer <- cbcl4_cumulative %>%
  pivot_longer(-cumulative_adv, names_to = "variables", values_to = "value")

#run anovas for 2 year cbcl
anovas_cbcl_2 <- cbcl2_cumulative.longer %>% 
  nest(data=(c(cumulative_adv, value))) %>% 
  mutate(model = map(data, ~anova(lm(value ~ cumulative_adv, .))), 
         tidy = map(model, broom::tidy)) %>% 
dplyr::select(variables, tidy) %>% 
unnest(cols=c(tidy)) 

anovas_cbcl_4 <- cbcl4_cumulative.longer %>% 
  nest(data=(c(cumulative_adv, value))) %>% 
  mutate(model = map(data, ~anova(lm(value ~ cumulative_adv, .))), 
         tidy = map(model, broom::tidy)) %>% 
dplyr::select(variables, tidy) %>% 
unnest(cols=c(tidy)) 

#adjust p-values
anovas_cbcl_2$BH <- p.adjust(test$p.value, method="BH")
anovas_cbcl_4$BH <- p.adjust(test$p.value, method="BH")

```
**Adversities Associated with SEF Domains**
Age 2 Years:

- Internalizing Problems: preconception adversity, prenatal adversity
- Total Problems: preconception adversity, prenatal adversity
- Developmental Problems: prenatal adversity

Other domains were also associated with adversities (e.g., anxious-depressed problems), but since these could be subsumed within internalizing symptoms, we chose to examine internalizing problems only. 

Age 4 Years:

- ADHD Problems: preconception adversity, prenatal adversity
- Attention Problems: preconception adversity, prenatal adversity
- Developmental Problems: prenatal adversity
- Externalizing Problems: preconception adversity, prenatal adversity
- Internalizing Problems: prenatal adversity
- Sleep Problems: prenatal adversity
- Total Problems: preconception adversity, prenatal adversity

In the same manner as for Age 2 Years variables, we selected the broadest child SEF domains from significant results to test for associations with gut microbiome composition. 

## Differential Abundance
### SEF at Age 2 years -- Without Covariates

*Note:* Covariates will be chosen based on overlap in these results (at the genus level) with the DA analyses for possible covariates that were run in the "Covariate Selection" chunks above.  

The same process as described in the adversity differential abundance sections is followed, in which analyses are run once with all cases (to get coefficient estimates) and once with complete cases (to get correct N and N.not.0). The same Maaslin2 default parameters are used as well.

#### Internalizing Problems
```{r eval=FALSE}
#INTERNALIZING PROBLEMS
#get dataset that is equivalent to one used for analyses with covariates
int2_metadata <- M24_metadata %>% 
  filter(!is.na(cbclintprobtot_m24_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary_ch)) #N=171
write_tsv(int2_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_intM24.tsv")
int2_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_intM24.tsv"

# Internalizing problems - 0.5% filtering threshold
maaslin_int_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = int2_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/no_covs/int_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("cbclintprobtot_m24_std", "ctq_total_score_25it", "STAI_prorated_state_pw26"), #correlated adversity variables are included
  standardize = TRUE
)

#adjust p-values 
int2_adjust <- maaslin_int_0.5p_nocovs$results %>% 
  filter(metadata=="cbclintprobtot_m24_std" & N.not.zero > (172*.15))

int2_adjust$qval = p.adjust(int2_adjust$pval, method="BH")
```

#### Developmental Problems
```{r eval=FALSE}
#DEVELOPMENTAL PROBLEMS
dev2_metadata <- M24_metadata %>% 
  filter(!is.na(cbcldevprobtot_m24_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(deliv_mode_str) & !is.na(only_bf_months) & !is.na(Sequencing_batch)) #N=166
write_tsv(dev2_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_devM24.tsv")
dev2_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_devM24.tsv" 

#developmental problems - 0.5% filtering threshold
maaslin_dev2_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = dev2_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/no_covs/dev_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm",
  fixed_effects = c("cbcldevprobtot_m24_std", "STAI_prorated_state_pw26"),
  standardize = TRUE
)
```

#### Total Problems
```{r eval=FALSE}
#TOTAL PROBLEMS
#create complete cases dataset to determine N and N.not.0
tot2_metadata <- M24_metadata %>% 
  filter(!is.na(cbcltotprobtot_m24_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(Sequencing_batch) & !is.na(ctq_total_score_25it)) #N=150
write_tsv(tot2_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_totM24.tsv")
tot2_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_totM24.tsv" 

#Total problems - 0.5% filtering threshold
maaslin_tot_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = tot2_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/no_covs/tot_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("cbcltotprobtot_m24_std", "ctq_total_score_25it", "STAI_prorated_state_pw26"),
  standardize = TRUE 
)
```
**Selected covariates:**  

* Internalizing problems at age 2 years: none  
* Total problems at age 2 years: sequencing batch   
* Developmental at age 2 years: delivery mode, duration of exclusive breastfeeding, child ethnicitiy  
*A priori covariates:* Child sex, any associated adversities  

### SEF at Age 4 years -- Without Covariates
#### ADHD Problems
```{r eval=FALSE}
#ADHD PROBLEMS
adhd4_metadata <- M24_metadata %>% 
  filter(!is.na(cbcladhdprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(ctq_total_score_25it)) #N=201
write_tsv(adhd4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_adhdY4.tsv")
adhd4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_adhdY4.tsv" 

#ADHD problems - 0.5% filtering threshold
maaslin_adhd4_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = adhd4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/no_covs/adhd_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25,
  analysis_method = "lm", 
  fixed_effects = c("cbcladhdprobtot_y4_std", "STAI_prorated_state_pw26", "ctq_total_score_25it"),
  standardize = TRUE 
)
```

#### Attention Problems
```{r eval=FALSE}
#ATTENTION PROBLEMS
attn4_metadata <- M24_metadata %>% 
  filter(!is.na(cbclattprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(ctq_total_score_25it)) #N=202
write_tsv(attn4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_attnY4.tsv")
attn4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_attnY4.tsv" 

#Attn problems - 0.5% filtering threshold
maaslin_att4_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = attn4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/no_covs/attn_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbclattprobtot_y4_std", "STAI_prorated_state_pw26", "ctq_total_score_25it"),
  standardize = TRUE 
)
```

#### Developmental Problems
```{r eval=FALSE}
#DEVELOPMENTAL PROBLEMS
#create complete cases dataset to determine N and N.not.0
dev4_metadata <- M24_metadata %>% 
  filter(!is.na(cbcldevprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary)) #N=289
write_tsv(dev4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_devY4.tsv")
dev4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_devY4.tsv" 

#Dev problems - 0.5% filtering threshold
maaslin_dev4_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = dev4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/no_covs/dev_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcldevprobtot_y4_std", "STAI_prorated_state_pw26"),
  standardize = TRUE 
)
```

#### Externalizing Problems
```{r eval=FALSE}
#EXTERNALIZING PROBLEMS
ext4_metadata <- M24_metadata %>% 
  filter(!is.na(cbclattprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(ctq_total_score_25it)) #N=202
write_tsv(ext4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_extY4.tsv")
ext4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_extY4.tsv" 

#Externalizing - 0.5% filtering threshold
maaslin_ext4_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = ext4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/no_covs/ext_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25,
  analysis_method = "lm", #default
  fixed_effects = c("cbclextprobtot_y4_std", "STAI_prorated_state_pw26", "ctq_total_score_25it"),
  standardize = TRUE 
  
)
```

#### Internalizing Problems
```{r eval=FALSE}
#INTERNALIZING PROBLEMS
#create complete cases dataset to determine N and N.not.0
int4_metadata <- M24_metadata %>% 
  filter(!is.na(cbclintprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(only_bf_months) & !is.na(Sequencing_batch) & !is.na(mother_ethnicity) & !is.na(Fibre.per.1000kcal)) #N=282
write_tsv(int4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_intY4.tsv")
int4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_intY4.tsv" 

#Internalizing problems - 0.5% filtering threshold
maaslin_int4_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = int4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/no_covs/int_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("cbclintprobtot_y4_std", "STAI_prorated_state_pw26"),
  standardize = TRUE
)
```

#### Sleep Problems
```{r eval=FALSE}
#SLEEP PROBLEMS
#create complete cases dataset to determine N and N.not.0
sleep4_metadata <- M24_metadata %>% 
  filter(!is.na(cbclattprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(only_bf_months) & !is.na(mother_ethnicity) & !is.na(probiotics_M24) & !is.na(Sequencing_batch)) #N=276
write_tsv(sleep4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_sleepY4.tsv")
sleep4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_sleepY4.tsv" 

#Sleep problems - 0.5% filtering threshold
maaslin_sleep4_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = sleep4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/no_covs/sleep_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbclsleepprobtot_y4_std", "STAI_prorated_state_pw26"),
  standardize = TRUE 
)
```

#### Total Problems
```{r eval=FALSE}
#TOTAL PROBLEMS
tot4_metadata <- M24_metadata %>% 
  filter(!is.na(cbcltotprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(ctq_total_score_25it)) #N=288
write_tsv(tot4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_totY4.tsv")
tot4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_totY4.tsv" 

#Total problems - 0.5% filtering threshold
maaslin_tot4_0.5p_nocovs = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = tot4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/no_covs/tot_raw_0.5p_results_nocovs_adversities_covsample",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("cbcltotprobtot_y4_std", "STAI_prorated_state_pw26", "ctq_total_score_25it"),
  standardize = TRUE 
)
```
**Selected covariates:**  

* adhd problems y4: none  
* attention probelms y4: none  
* developmental problems y4: none  
* externalizing problems y4: none  
* internalizing problems y4: duration of exclusive breastfeeding, child ethnicity, sequencing batch, fiber (Ruminoclostridium overlap)
* sleep problems y4: duration of exclusive breastfeeding, child ethnicity, probiotic use, sequencing batch  
* total problems y4: none  
*A prior covariates:* child sex, any associated adversities  


### SEF at Age 2 years -- Including Covariates
The same process as described in the adversity differential abundance sections is followed, in which analyses are run once with all cases (to get coefficient estimates) and once with complete cases (to get correct N and N.not.0). The same Maaslin2 default parameters are used as well.
#### Developmental Problems
```{r eval=FALSE}
#DEVELOPMENTAL PROBLEMS
dev2_metadata <- M24_metadata %>% 
  filter(!is.na(cbcldevprobtot_m24_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(deliv_mode_str) & !is.na(only_bf_months) & !is.na(Sequencing_batch)) #N=166
write_tsv(dev2_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_devM24.tsv")
dev2_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_devM24.tsv" 

#developmental problems - 1% filtering threshold
maaslin_dev2_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/dev_raw_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcldevprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch", "deliv_mode_str", "only_bf_months", "Sequencing_batch"),
  standardize = TRUE
) 

#developmental problems - 0.5% filtering threshold
maaslin_dev2_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/dev_raw_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcldevprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch", "deliv_mode_str", "only_bf_months", "Sequencing_batch"),
  standardize = TRUE
) 
  
#developmental problems - 0.1% filtering threshold
maaslin_dev2_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/dev_raw_0.1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcldevprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch", "deliv_mode_str", "only_bf_months", "Sequencing_batch"),
  standardize = TRUE 
)
```


#### Internalizing Problems
```{r eval=FALSE}
#INTERNALIZING PROBLESM
#create metadata file with complete cases for internalizing
int2_metadata <- M24_metadata %>% 
  filter(!is.na(cbclintprobtot_m24_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary)) #N=171
write_tsv(int2_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_intM24.tsv")
int2_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_intM24.tsv"

# Internalizing problems - 0.5% filtering threshold
maaslin_int_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/int_raw_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbclintprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch"),
  standardize = TRUE 
)

# Internalizing problems - 1% filtering threshold
maaslin_int_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/int_raw_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25,
  analysis_method = "lm", #default
  fixed_effects = c("cbclintprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch"),
  standardize = TRUE
)

# Internalizing problems - 0.1% filtering threshold
maaslin_int_0.1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/int_raw_0.1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbclintprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch"),
  standardize = TRUE
)
```

#### Total Problems
```{r eval=FALSE}
#TOTAL PROBLEMS
#create metadata file with complete cases for total
tot2_metadata <- M24_metadata %>% 
  filter(!is.na(cbcltotprobtot_m24_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(Sequencing_batch) & !is.na(ctq_total_score_25it)) #N=150
write_tsv(tot2_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_totM24.tsv")
tot2_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_totM24.tsv" 

#Total problems - 0.5% filtering threshold
maaslin_tot_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/tot_raw_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcltotprobtot_m24_std", "ctq_total_score_25it", "STAI_prorated_state_pw26", "sex_binary_ch", "Sequencing_batch"),
  standardize = TRUE
)

#Total problems - 1% filtering threshold
maaslin_tot_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/tot_raw_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcltotprobtot_m24_std", "ctq_total_score_25it", "STAI_prorated_state_pw26", "sex_binary_ch", "Sequencing_batch"),
  standardize = TRUE 
)

#Total problems - 0.1% filtering threshold
maaslin_tot_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/tot_raw_0.1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcltotprobtot_m24_std", "ctq_total_score_25it", "STAI_prorated_state_pw26", "sex_binary_ch", "Sequencing_batch"),
  standardize = TRUE 
)
```

### SEF at Age 4 years -- Including Covariates
The same process as described in the adversity differential abundance sections is followed, in which analyses are run once with all cases (to get coefficient estimates) and once with complete cases (to get correct N and N.not.0). The same Maaslin2 default parameters are used as well.
#### ADHD Problems
```{r eval=FALSE}
#ADHD PROBLEMS
adhd4_metadata <- M24_metadata %>% 
  filter(!is.na(cbcladhdprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(ctq_total_score_25it)) #N=201
write_tsv(adhd4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_adhdY4.tsv")
adhd4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_adhdY4.tsv" 

#ADHD problems - 0.5% filtering threshold
maaslin_adhd4_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/adhd_raw_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbcladhdprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#ADHD problems - 1% filtering threshold
maaslin_adhd4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/adhd_raw_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm",
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbcladhdprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#ADHD problems - 0.1% filtering
maaslin_adhd4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/adhd_raw_0.1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbcladhdprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)
```

#### Attention Problems
```{r eval=FALSE}
#ATTENTION PROBLEMS
attn4_metadata <- M24_metadata %>% 
  filter(!is.na(cbclattprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(ctq_total_score_25it)) #N=202
write_tsv(attn4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_attnY4.tsv")
attn4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_attnY4.tsv" 

#Attention problems - 0.5% filtering threshold
maaslin_att4_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/attn_raw_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbclattprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#Attention problems - 1% filtering threshold
maaslin_att4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/attn_raw_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbclattprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#Attention problems - 0.1% filtering threshold
maaslin_att4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/attn_raw_0.1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbclattprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)
```

#### Developmental Problems
```{r eval=FALSE}
#DEVELOPMENTAL PROBLEMS
dev4_metadata <- M24_metadata %>% 
  filter(!is.na(cbcldevprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary)) #N=289
write_tsv(dev4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_devY4.tsv")
dev4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_devY4.tsv" 

#Developmental problems - 0.5% filtering threshold
maaslin_dev4_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/dev_raw_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "cbcldevprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#Developmental problems - 0.1% filtering threshold
maaslin_dev4_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = dev4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/dev_raw_0.1p_results_completecases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "cbcldevprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#Developmental problems - 1% filtering threshold
maaslin_dev4_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = dev4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/dev_raw_0.1p_results_completecases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "cbcldevprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

```

#### Externalizing Problems
```{r eval=FALSE}
#EXTERNALIZING PROBLEMS
ext4_metadata <- M24_metadata %>% 
  filter(!is.na(cbclattprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(ctq_total_score_25it)) #N=202
write_tsv(ext4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_extY4.tsv")
ext4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_extY4.tsv" 

#Externalizing - 0.5% filtering threshold
maaslin_ext4_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/ext_raw_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbclextprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#Externalizing - 1% filtering threshold
maaslin_ext4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/ext_raw_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbclextprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#Externalizing - 0.1% filtering threshold
maaslin_ext4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/ext_raw_0.1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbclextprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)
```

#### Internalizing Problems
```{r eval=FALSE}
#INTERNALIZING PROBLEMS
#get complete cases metadata file
int4_metadata <- M24_metadata %>% 
  filter(!is.na(cbclintprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(only_bf_months) & !is.na(Sequencing_batch) & !is.na(mother_ethnicity) & !is.na(Fibre.per.1000kcal)) #N=282
write_tsv(int4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_intY4.tsv")
int4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_intY4.tsv" 

#Internalzing problems - 0.5% filtering threshold
maaslin_int4_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/int_raw_0.5p_results_compcases_fiber_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "cbclintprobtot_y4_std", "sex_binary_ch", "only_bf_months", "mother_ethnicity", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)'
)

#Internalzing problems - 1% filtering threshold
maaslin_int4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = int4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/int_raw_1p_results_compcases_fiber",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "cbclintprobtot_y4_std", "sex_binary_ch", "only_bf_months", "mother_ethnicity", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)'
)

#Internalzing problems - 0.1% filtering threshold
maaslin_int4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = int4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/int_raw_0.1p_results_compcases_fiber",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "cbclintprobtot_y4_std", "sex_binary_ch", "only_bf_months", "mother_ethnicity", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)'
)
```

#### Sleep Problems
```{r eval=FALSE}
#SLEEP PROBLEMS
sleep4_metadata <- M24_metadata %>% 
  filter(!is.na(cbclattprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) & !is.na(only_bf_months) & !is.na(mother_ethnicity) & !is.na(probiotics_M24) & !is.na(Sequencing_batch)) #N=276
write_tsv(sleep4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_sleepY4.tsv")
sleep4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_sleepY4.tsv" 

#Sleep problems - 0.5% filtering threshold
maaslin_sleep4_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/sleep_raw_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "cbclsleepprobtot_y4_std", "sex_binary_ch", "only_bf_months", "mother_ethnicity", "probiotics_M24", "Sequencing_batch"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)'
)

#Sleep problems - 1% filtering threshold
maaslin_sleep4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/sleep_raw_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "cbclsleepprobtot_y4_std", "sex_binary_ch", "only_bf_months", "mother_ethnicity", "probiotics_M24", "Sequencing_batch"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)'
)

#Sleep problems - 0.1% filtering threshold
maaslin_sleep4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/sleep_raw_0.1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "cbclsleepprobtot_y4_std", "sex_binary_ch", "only_bf_months", "mother_ethnicity", "probiotics_M24", "Sequencing_batch"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)'
)
```

#### Total Problems
```{r eval=FALSE}
#TOTAL PROBLEMS
tot4_metadata <- M24_metadata %>% 
  filter(!is.na(cbcltotprobtot_y4_std) & !is.na(STAI_prorated_state_pw26) & !is.na(sex_binary) ) #N=288
write_tsv(tot4_metadata, "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_totY4.tsv")
tot4_metadata_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/metadata/metadata_totY4.tsv" 


#Total problems - 0.5% filtering threshold
maaslin_tot4_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/tot_raw_0.5p_results_allcases_r1",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbcltotprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#Total problems - 1% filtering threshold
maaslin_tot4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/tot_raw_1p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.01,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbcltotprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

#Total problems - 0.1% filtering threshold
maaslin_tot4_1p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = tot4_metadata_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/tot_raw_0.1p_results_completecases",
  min_abundance = 0,
  min_prevalence = 0.001,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbcltotprobtot_y4_std", "sex_binary_ch"),
  standardize = TRUE 
)

```

### Consolidate main differential abundance results into one dataset 
For graphing below  
Only including taxa with N.not.0 >= 15% of all samples in the analysis, and datasets at 0.5% filtering threshold
```{r eval=FALSE}
#Combine Adversity Results
#find cutoff Ns for each analysis
ctq_n <- alpha_diversity_stress_usable %>% dplyr::count(!is.na(ctq_total_score_25it) & !is.na(mother_ethnicity) & !is.na(only_bf_months) & !is.na(Sequencing_batch) & !is.na(Fibre.per.1000kcal)) 
stai_n <- alpha_diversity_stress_usable %>% dplyr::count(!is.na(STAI_prorated_state_pw26) & !is.na(mother_ethnicity) & !is.na(only_bf_months) & !is.na(Sequencing_batch) & !is.na(probiotics_M24) & !is.na(Fibre.per.1000kcal)) 
leq_n <- alpha_diversity_stress_usable %>% dplyr::count(!is.na(postnatal_stress) & !is.na(mother_ethnicity)) 
all_adv <- alpha_diversity_stress_usable %>% dplyr::count(!is.na(postnatal_stress) & !is.na(ctq_total_score_25it) & !is.na(STAI_prorated_state_pw26) & !is.na(mother_ethnicity) &!is.na(only_bf_months)&!is.na(Sequencing_batch) &!is.na(c_age_years_stoolproxy_m24))
cumu_n <-  alpha_diversity_stress_usable %>% dplyr::count(!is.na(cumulative_adv))

#grab results of interest from the all cases run (so N and N.not.0 are incorrect, but coefficient estimates are correct)
ctq_results <- maaslin_maltreatment_covs_0.5p$results %>% 
  dplyr::filter(metadata=="ctq_total_score_25it" & qval < .25 & N.not.zero >= .15*ctq_n$n[2]) %>% 
  mutate(
    analysis = "preconception"
  )
stai_results <- maaslin_prenatanx_covs_0.5p$results %>% 
  dplyr::filter(metadata=="STAI_prorated_state_pw26" & qval < .25 & N.not.zero >= .15*stai_n$n[2]) %>% 
  mutate(
    analysis = "prenatal"
  )
leq_results <- maaslin_poststress_covs_0.5p$results %>% 
  dplyr::filter(metadata=="postnatal_stress_s" & qval < .25 & N.not.zero >= .15*leq_n$n[2]) %>% 
  mutate(
    analysis = "postnatal"
  )
alladv_results <- maaslin2_alladv_withcov_05$results %>% 
  dplyr::filter((metadata=="ctq_total_score_25it" | metadata=="STAI_prorated_state_pw26" | metadata=="postnatal_stress_s") & qval < .25 & N.not.zero >=.15*all_adv$n[2]) %>% 
  mutate(
    analysis = "all_adv"
  )
#no significant results for cumulative, so do not include here

adversity_results <- ctq_results %>% 
  bind_rows(stai_results) %>% 
  bind_rows(leq_results) %>% 
  bind_rows(alladv_results)

#filter out the prenatal adversity Streptococcus.agalactiae.18RS21 as the N is actually too small in complete case analyses (N=276, N.not.0 = 25)
adversity_results <- adversity_results %>% 
  filter(N.not.zero != 54)

#Combine Socioemotional Functioning Results
#Ns hardcoded from complete cases analyses to save space

#grab results of interest from all cases run (so N and N.not.0 are incorrect, but coefficient estimates are correct)
maaslin_int2_results <- maaslin_int_0.5p$results %>%
  dplyr::filter(metadata=="cbclintprobtot_m24_std" & qval < .25 & N.not.zero >= .15*140)  %>%
  dplyr::mutate(analysis = "internalizing symptoms at age 2 years")

maaslin_dev2_results <- maaslin_dev2_0.5p$results %>% 
  dplyr::filter(metadata=="cbcldevprobtot_m24_std" & qval < .25 & N.not.zero >= .15*166)  %>% 
  dplyr::mutate(analysis = "developmental problems at age 2 years")

maaslin_tot2_results <- maaslin_tot_0.5p$results %>% 
  dplyr::filter(metadata=="cbcltotprobtot_m24_std" & qval < .25 & N.not.zero >= .15*150)  %>% 
  dplyr::mutate(analysis = "total problems at age 2 years")

maaslin_tot4_results <- maaslin_tot4_0.5p$results %>%
  dplyr::filter(metadata=="cbcltotprobtot_y4_std" & qval < .25 & N.not.zero >= .15*289)  %>%
  dplyr::mutate(analysis = "total problems at age 4 years")

maaslin_att4_results <- maaslin_att4_0.5p$results %>%
  dplyr::filter(metadata=="cbclattprobtot_y4_std" & qval < .25 & N.not.zero >= .15*202)  %>%
  dplyr::mutate(analysis = "attention problems at age 4 years")

maaslin_ext4_results <- maaslin_ext4_0.5p$results %>%
  dplyr::filter(metadata=="cbclextprobtot_y4_std" & qval < .25 & N.not.zero >= .15*202)  %>%
  dplyr::mutate(analysis = "externalizing problems at age 4 years")

maaslin_int4_results <- maaslin_int4_0.5p$results %>% 
  dplyr::filter(metadata=="cbclintprobtot_y4_std" & qval < .25 & N.not.zero >= .15*282)  %>% 
  dplyr::mutate(analysis = "internalizing problems at age 4 years")

maaslin_sleep4_results <- maaslin_sleep4_0.5p$results %>% 
  dplyr::filter(metadata=="cbclsleepprobtot_y4_std" & qval < .25 & N.not.zero >= .15*276)  %>% 
  dplyr::mutate(analysis = "sleep problems at age 4 years")

maaslin_dev4_results <- maaslin_dev4_0.5p$results %>%
  dplyr::filter(metadata=="cbcldevprobtot_y4_std" & qval < .25 & N.not.zero >= .15*289)  %>%
  dplyr::mutate(analysis = "dev problems at age 4 years")

maaslin_adhd4_results <- maaslin_adhd4_0.5p$results %>%
  dplyr::filter(metadata=="cbcladhdtot_y4_std" & qval < .25 & N.not.zero >= .15*202)  %>%
  dplyr::mutate(analysis = "adhd problems at age 4 years")

maaslin_.5_results <-
  adversity_results %>% 
  bind_rows(maaslin_tot2_results) %>% 
  bind_rows(maaslin_dev2_results) %>% 
  bind_rows(maaslin_int2_results) %>% 
  bind_rows(maaslin_tot4_results) %>% 
  bind_rows(maaslin_ext4_results) %>% 
  bind_rows(maaslin_int4_results) %>% 
  bind_rows(maaslin_sleep4_results) %>% 
  bind_rows(maaslin_dev4_results) %>% 
  bind_rows(maaslin_att4_results) %>% 
  bind_rows(maaslin_adhd4_results)

#Create new variables for graphing
#create new var: log2(exp(coeff))
maaslin_.5_results <-
  maaslin_.5_results %>% 
  dplyr::rename(taxa = feature) %>% 
  dplyr::mutate(
    log2FoldChange = sign(coef)*log(2*(exp(abs(coef)))),
    se_min = coef-stderr,
    se_max = coef+stderr,
    semin_log2fc = sign(se_min)*log(2*(exp(se_min))),
    semax_log2fc = sign(se_max)*log(2*(exp(se_max))),
    error_margin = qt(0.975, N.not.zero - 2, lower.tail = TRUE) * stderr, # manually calculate 95% confidence interval for a regression coefficient (t-value for alpha/2, df-2 x SE)
    lower_CI = coef - error_margin,
    upper_CI = coef + error_margin,
    lCI_log2fc = sign(lower_CI)*log(2*(exp(abs(lower_CI)))),
    uCI_log2fc = sign(upper_CI)*log(2*(exp(abs(upper_CI)))),
    fact_analysis = factor(analysis)
  )

#create new var with shorter taxa names
maaslin_.5_results <-
  maaslin_.5_results %>% 
  dplyr::mutate(
    Taxa = case_when(str_detect(taxa, "Intestinibacter") ~ "g__Intestinibacter",
                     str_detect(taxa, "sensu.stricto") ~ "g__Clostridium sensu stricto",
                     str_detect(taxa, "Streptococcus") ~ "g__Streptococcus",
                     str_detect(taxa, "Parabacteroides") ~ "g__Parabacteroides",
                     str_detect(taxa, "Finegoldia") ~ "g__Finegoldia",
                     str_detect(taxa, "Ruminococcus") ~ "g__Ruminococcus",
                     str_detect(taxa, "Blautia") ~ "g__Blautia",
                     str_detect(taxa, "Peptoclostridium") ~ "g__Peptoclostridium",
                     str_detect(taxa, "Veillonella") ~ "g__Veillonella",
                     str_detect(taxa, "UCG") ~ "f__Lachnospiraceae",
                     str_detect(taxa, "Faecalibacterium") ~ "g__Faecalibacterium",
                     str_detect(taxa, "Coprobacillus") ~ "g__Coprobacillus"
    )
  )

#create more specific analysis variable that specifies the significant predictor in the models with all adversities together
maaslin_.5_results <- maaslin_.5_results %>% 
  dplyr::mutate(
    analysis = case_when(
      analysis=="all_adv" & name=="STAI_prorated_state_pw26" ~ "prenatal (all adversities in model)",
      analysis=="all_adv" & name=="postnatal_stress_syes" ~ "postnatal (all adversities in model)",
      analysis != "all_adv" ~ analysis
      )
  )

write_csv(maaslin_.5_results, "../../data/analysis_datasets/fran/maaslin2/final_results/all_maaslin_results_0.5p_r1.csv")
```

## Differential Abundance Dotplots
### Figure 2: Differentially Abundant Taxa as a Function of Adversity Exposure.
```{r dotplot adversity, warning=FALSE}
maaslin_.5_results_figures <- read_csv("../../data/analysis_datasets/fran/maaslin2/final_results/all_maaslin_results_0.5p_r1_fixed.csv")

#add 'species' variable to deal with two Strepto results with same advresity exposure not being separated by position_dodge (for errorbars and points)
maaslin_.5_results_figures <- maaslin_.5_results_figures %>% 
  mutate(species = case_when( #set one of the results with duplicate genuses to have the species be '2' while the other entry for that genus is '1'
    Taxa=="g__Streptococcus" & analysis=="prenatal" & N.not.zero==249 ~ 2,
    Taxa=="g__Parabacteroides" & analysis=="postnatal" ~ 2,
    Taxa=="g__Ruminococcus" & analysis=="prenatal" ~ 2,
    TRUE ~ 1))

#create dotplot for adversity variables
ggplot(data=maaslin_.5_results_figures %>% filter(str_detect(analysis, "preconception|prenatal|postnatal")), aes(x = Taxa, y = log2FoldChange, ymin=lCI_log2fc, ymax=uCI_log2fc, color=factor(analysis), fill=factor(analysis), shape = factor(analysis), group=species)) +
  geom_point(aes(color=factor(analysis), fill=factor(analysis), shape = factor(analysis)), size = 2, position=position_dodge(width=0.3)) +
  geom_errorbar(aes(color=factor(analysis), fill=factor(analysis), shape = factor(analysis)), width=0, position=position_dodge(width=0.3)) +
  scale_x_discrete(expand=c(0.2, 0)) +
  coord_flip() +
  scale_y_reverse(limits = -1.5, 1.5) + 
  #ylim(-1.5, 1.5) + #needs to be ylim bc coordinates have been flipped
  scale_y_continuous(breaks=seq(-1.5, 1.5, by = 0.50)) +
  geom_hline(aes(yintercept = 0)) + 
  scale_shape_manual(values=c(21, 22, 23, 24, 25)) + #specify specific shapes (easiest to see)
  scale_fill_brewer(palette = "Dark2") + #specify colorblind friendly color palette for fill
  scale_color_brewer(palette = "Dark2") + #specify colorblind friendly color palette for color
  labs(color = "Adversity Exposure", shape = "Adversity Exposure", fill = "Adversity Exposure") + #change legend title 
  theme(axis.text = element_text(size = 10)) +
  theme_minimal() 
  
ggsave("../../manuscripts/fran_adversity_microbiome/figures/adversity_da_figure_r1.svg", height = 4, width = 7, bg = "white", dpi=1000)
```       

### Figure 3: Differentially Abundant Taxa as a Function of Adversity-Correlated Child Socioemotional Functioning
```{r dotplot SEF, warning=FALSE}
#create dotplot w taxa for SEF variables
ggplot(data=maaslin_.5_results_figures %>% filter(str_detect(name, "cbcl")), aes(x = Taxa, y = log2FoldChange, color=factor(analysis))) +
  geom_point(aes(color = factor(analysis), shape = factor(analysis), fill=factor(analysis)), size = 5, position=position_dodge(width=0.4)) +
  geom_errorbar(aes(ymin=lCI_log2fc, ymax=uCI_log2fc), width=0, position=position_dodge(width=0.4)) +
  scale_x_discrete(expand=c(0.2, 0)) +
  coord_flip() +
  scale_y_reverse(limits = -1.5, 1.5) + #needs to be ylim bc coordinates have been flipped
  scale_y_continuous(breaks=seq(-1.5, 1.5, by = 0.50)) + 
  geom_hline(aes(yintercept = 0)) + 
  scale_shape_manual(values=c(21, 22, 23, 24, 25)) + #specify specific shapes (easiest to see)
  scale_fill_brewer(palette = "Dark2") + #specify colorblind friendly color palette for fill
  scale_color_brewer(palette = "Dark2") + #specify colorblind friendly color palette for color
  labs(color = "Socioemotional Functioning Domain", shape = "Socioemotional Functioning Domain", fill = "Socioemotional Functioning Domain") + #change legend title 
  theme(axis.text = element_text(size = 11)) +
  theme_minimal()
  
ggsave("../../manuscripts/fran_adversity_microbiome/figures/SEF_da_figure_biggerpoints.svg", height = 4, width=7, bg="white", dpi=1000)
```

## Additional Analyses based on PNAS reviewer comments

### Evaluating missing data mechanism
MCAR -- missing completely at random vs. MAR -- missing (conditionally) at random
Suggestions to evaluate whether MCAR is potentially true in this dataset are based on Enders, C. K. (2010). Applied missing data analysis, 2nd edition. Guilford Press.
Raykov et al. (2012) proposes using BH for FDR correction
```{r}
#create missingness indicator variables for each adversity
alpha_diversity_stress_usable_missing <- alpha_diversity_stress_usable %>% 
  mutate(
    ctq_miss = as.factor(if_else(is.na(ctq_total_score_25it), 1, 0)),
    stai_miss = as.factor(if_else(is.na(STAI_prorated_state_pw26), 1, 0)),
    postnat_miss = as.factor(if_else(is.na(postnatal_stress), 1, 0)),
    income_miss = as.factor(if_else(is.na(monthly_income_per_member), 1, 0)
  ))

#univariate pattern mean differences -- test whether people misisng vs. not missing CTQ, postnatal stress, prenatal stress have different average: levels of other adversities, income, alpha diversity (t-tests)
mcar_testing <- alpha_diversity_stress_usable_missing %>%
  dplyr::select(
    ctq_miss,
    stai_miss,
    postnat_miss,
    ctq_total_score_25it,
    STAI_prorated_state_pw26,
    postnatal_stress,
    monthly_income_per_member,
    shannon_entropy,
    faith_pd,
    pielou_evenness,
    observed_features
  ) 
#do the t-tests for ctq missing
test<-lapply(mcar_testing_ctq[,c("STAI_prorated_state_pw26", "monthly_income_per_member", "shannon_entropy", "faith_pd", "pielou_evenness", "observed_features")], function(x) t.test(x ~ mcar_testing$ctq_miss))
#adjust p-values 
p.adjust(c(test$STAI_prorated_state_pw26$p.value, test$monthly_income_per_member$p.value, test$shannon_entropy$p.value, test$faith_pd$p.value, test$pielou_evenness$p.value, test$observed_features$p.value), method = "BH")

#variance tests for ctq missing
test_var <- lapply(mcar_testing_ctq[,c("STAI_prorated_state_pw26", "monthly_income_per_member", "shannon_entropy", "faith_pd", "pielou_evenness", "observed_features")], function(x) var.test(x ~ mcar_testing$ctq_miss))

p.adjust(c(test_var$STAI_prorated_state_pw26$p.value, test_var$monthly_income_per_member$p.value, test_var$shannon_entropy$p.value, test_var$faith_pd$p.value, test_var$pielou_evenness$p.value, test_var$observed_features$p.value), method = "BH")

#do the t-tests for stai missing
test2 <- lapply(mcar_testing_ctq[,c("ctq_total_score_25it", "monthly_income_per_member", "shannon_entropy", "faith_pd", "pielou_evenness", "observed_features")], function(x) t.test(x ~ mcar_testing$stai_miss))

p.adjust(c(test2$ctq_total_score_25it$p.value, test2$monthly_income_per_member$p.value, test2$shannon_entropy$p.value, test2$faith_pd$p.value, test2$pielou_evenness$p.value, test2$observed_features$p.value), method = "BH")

#variance testing for stai missing
test2_var <- lapply(mcar_testing_ctq[,c("ctq_total_score_25it", "monthly_income_per_member", "shannon_entropy", "faith_pd", "pielou_evenness", "observed_features")], function(x) var.test(x ~ mcar_testing$stai_miss))

p.adjust(c(test2_var$ctq_total_score_25it$p.value, test2_var$monthly_income_per_member$p.value, test2_var$shannon_entropy$p.value, test2_var$faith_pd$p.value, test2_var$pielou_evenness$p.value, test2_var$observed_features$p.value), method = "BH")

#t-tests for postnatal stress missing
test3 <- lapply(mcar_testing_ctq[,c("ctq_total_score_25it", "STAI_prorated_state_pw26", "monthly_income_per_member", "shannon_entropy", "faith_pd", "pielou_evenness", "observed_features")], function(x) t.test(x ~ mcar_testing$postnat_miss))

p.adjust(c(test3$ctq_total_score_25it$p.value, test3$STAI_prorated_state_pw26$p.value, test3$monthly_income_per_member$p.value, test3$shannon_entropy$p.value, test3$faith_pd$p.value, test3$pielou_evenness$p.value, test3$observed_features$p.value), method = "BH")

#variance tests for postnatal stress missing
test3_var <- lapply(mcar_testing_ctq[,c("ctq_total_score_25it", "STAI_prorated_state_pw26", "monthly_income_per_member", "shannon_entropy", "faith_pd", "pielou_evenness", "observed_features")], function(x) var.test(x ~ mcar_testing$postnat_miss))

p.adjust(c(test3_var$ctq_total_score_25it$p.value, test3_var$STAI_prorated_state_pw26$p.value, test3_var$monthly_income_per_member$p.value, test3_var$shannon_entropy$p.value, test3_var$faith_pd$p.value, test3_var$pielou_evenness$p.value, test3_var$observed_features$p.value), method = "BH")

#chisquare tests with missing indicators and postnatal stress
mcar_test_chisq <- mcar_testing %>% 
  dplyr::select(
    ctq_miss,
    stai_miss,
    postnatal_stress
  ) %>% drop_na()

chisq.test(mcar_test_chisq$ctq_miss, mcar_test_chisq$postnatal_stress) #ns
chisq.test(mcar_test_chisq$stai_miss, mcar_test_chisq$postnatal_stress) #ns

#little's mcar test
library(misty)
mcar_testing_noind <- mcar_testing %>% 
  dplyr::select(
    -contains("miss"),
    -postnatal_stress #remove categorical variable because little's test is not suitable for categorical variables
  )

na.test(mcar_testing_noind)
``` 

### Ethnicity
Switch to treating ethnicity as an a priori covariate
#### Alpha diversity
```{r}
#all exposures

#models
ethadd_shannon_mult <- lm(shannon_entropy~sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch+mother_ethnicity+ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress, data = alpha_diversity_stress_usable)
ethadd_faith_mult <- lm(faith_pd~sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch+mother_ethnicity+ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress, data = alpha_diversity_stress_usable)
ethadd_feat_mult <- lm(observed_features~sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch+mother_ethnicity+ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress, data = alpha_diversity_stress_usable)
ethadd_even_mult <- lm(pielou_evenness~sex_binary+deliv_mode+monthly_income_per_member+Sequencing_batch+mother_ethnicity+ctq_total_score_25it+STAI_prorated_state_pw26+postnatal_stress, data = alpha_diversity_stress_usable)

#results
tab_model(ethadd_shannon_mult, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/shannon_ethadd.doc") #NOTE: this function automatically does dummy coding with chinese as reference group (confirmed results were same as with dummy coded factors)
tab_model(ethadd_faith_mult, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/faith_ethadd.doc")
tab_model(ethadd_feat_mult, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/obsfeatures_ethadd.doc")
tab_model(ethadd_even_mult, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/evenness_ethadd.doc")

#cumulative adversity
ethadd_shannon_cumulative0v2 <- lm(shannon_entropy~cumulative_0v1 + cumulative_0v2+sex_binary+deliv_mode+mother_ethnicity+monthly_income_per_member+Sequencing_batch, data = alpha_diversity_stress_usable)
ethadd_faith_cumulative0v2 <- lm(faith_pd~cumulative_0v1 + cumulative_0v2+sex_binary+deliv_mode+mother_ethnicity+monthly_income_per_member+Sequencing_batch, data = alpha_diversity_stress_usable)
ethadd_feat_cumulative0v2 <- lm(observed_features~cumulative_0v1 + cumulative_0v2+sex_binary+deliv_mode+mother_ethnicity+monthly_income_per_member+Sequencing_batch, data = alpha_diversity_stress_usable)
ethadd_even_cumulative0v2 <- lm(pielou_evenness~cumulative_0v1 + cumulative_0v2+sex_binary+deliv_mode+mother_ethnicity+monthly_income_per_member+Sequencing_batch, data = alpha_diversity_stress_usable)

#results
tab_model(ethadd_shannon_cumulative0v2, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/shannon_cumulative_ethadd.doc") #NOTE: this function automatically does dummy coding with chinese as reference group (confirmed results were same as with dummy coded factors)
tab_model(ethadd_faith_cumulative0v2, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/faith_cumulative_ethadd.doc")
tab_model(ethadd_feat_cumulative0v2, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/obsfeatures_cumulative_ethadd.doc")
tab_model(ethadd_even_cumulative0v2, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE, file = "../../manuscripts/fran_adversity_microbiome/tables/alpha_diversity/evenness_cumulative_ethadd.doc")
```
No differences in results from covariates analyses without ethnicity

#### Differential abundance
Analyses controlling for ethnicity that haven't already controlled for ethnicity
```{r}
#filepaths to load again if needed:
M24_meta_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/M24_metadata.tsv"
tax_features_ra_fp <-  "../../data/analysis_datasets/fran/maaslin2/inputs/M24_tax_features_ra.tsv"

#cumulative adversity (0.5% filtering)
maaslin2_cumulative_eth_0.5 = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/cumulative_adversity/cumulative_ethadded_0.5p_results_allcases",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm",
  fixed_effects = c("cumulative_adv_s", "mother_ethnicity"),
  standardize = TRUE, 
  reference = c('cumulative_adv_s,0 tp', 'mother_ethnicity, chinese') 
)

#RESULTS:no sig bacteria by cumulative adversity (same as with no covariates)

#developmental problems y2 (0.5%) 
maaslin_dev2_ethadd_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/dev_raw_0.5p_results_allcases_ethadded",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcldevprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch", "deliv_mode_str", "only_bf_months", "Sequencing_batch", "mother_ethnicity"),
  standardize = TRUE,
  reference = 'mother_ethnicity, chinese' 
) 

#RESULTS: no differences in results from covariate analyses without ethnicity added 

#internalizing problems y2 (0.5% filtering threshold)
maaslin_int__eth_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/int_raw_0.5p_results_allcases_ethadded",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbclintprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch", "mother_ethnicity"),
  standardize = TRUE,
  reference = c('mother_ethnicity, chinese')
)

#RESULTS: no difference in results from covariate analysis without ethnicity added

#total problems y2
maaslin_tot_ethadd_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/tot_raw_0.5p_results_allcases_ethadded",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcltotprobtot_m24_std", "ctq_total_score_25it", "STAI_prorated_state_pw26", "sex_binary_ch", "Sequencing_batch", "mother_ethnicity"),
  standardize = TRUE,
  reference = c('mother_ethnicity, chinese')
)

#RESULTS: no difference

#adhd problems y4
maaslin_adhd4_ethadsd_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/adhd_raw_0.5p_results_allcases_ethadded",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbcladhdprobtot_y4_std", "sex_binary_ch", "mother_ethnicity"),
  standardize = TRUE,
  reference = c('mother_ethnicity, chinese')
)

#RESULTS: no difference

#attention problems y4
maaslin_att4_ethadd_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/attn_raw_0.5p_results_allcases_ethadded",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbclattprobtot_y4_std", "sex_binary_ch", "mother_ethnicity"),
  standardize = TRUE,
  reference = c('mother_ethnicity, chinese')
)

#RESULTS: no difference

#developmental problems y4
maaslin_dev4__ethadd_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/dev_raw_0.5p_results_allcases_ethadded",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "cbcldevprobtot_y4_std", "sex_binary_ch", "mother_ethnicity"),
  standardize = TRUE,
  reference = c('mother_ethnicity, chinese')
)

#RESULTS: no difference

#externalizing problems y4
maaslin_ext4_ethadd_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/ext_raw_0.5p_results_allcases_ethadded",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbclextprobtot_y4_std", "sex_binary_ch", "mother_ethnicity"),
  standardize = TRUE,
  reference = c('mother_ethnicity, chinese')
)

#RESULTS: no difference

#total problems y4
maaslin_tot4_ethadd_0.5p = Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/tot_raw_0.5p_results_allcases_ethadded",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "ctq_total_score_25it", "cbcltotprobtot_y4_std", "sex_binary_ch", "mother_ethnicity"),
  standardize = TRUE,
  reference = c('mother_ethnicity, chinese')
)

#RESULTS: one additional significant (Streptoccous) - q = .22 vs. q=.6 w/o ethnicity
```

### Do adversity exposures co-occur in the sample?
```{r}
#already looked at correlation between prenatal and preconception (Figure S2), r = .18, p < .001

#t-test with postnatal adversity group and preconception adversity
t.test(ctq_total_score_25it~postnatal_stress, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns

#t-test with postnatal adversity group and prenatal adversity
t.test(STAI_prorated_state_pw26~postnatal_stress, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
```

### What effect are the covariates having in associations where adding/removing covariates changes significance?
If significant but then becomes n.s. when covariates included: potentially collinearity too high (decreases power when both entered in model)
Analyses that differed with vs. without including covariates:
- postnatal adversity & parabacteriodes (all adv in model; n.s. when covariates not included)
- postnatal adversity & parabacteriodes (separate model; n.s. when covariates not included)
- postnatal adversity & finegoldia (separate model; n.s. when covariates not included)
- prenatal adversity & streptococcus (separate model; n.s. when covariates not included)
- developmental Y2 & XXX (separate model; n.s. when covariates not included)
- internalizing Y4 & XXX (separate model; n.s. when covariates not included)
- sleep Y4 & XXX (separate model; n.s. when covariates not included)
```{r}
#test whether selected alpha div covariates are significantly associated with adversity predictors
#from bivariate correlations: income, fiber, pufa & prenatal, precon adversity
  #income weakly negatively correlated with prenatal adversity (r = -0.14), income theorized to potentially be a confounder -- prenatal anxiety cannot cause differences in income, but diffs in income could influence anxiety and gm
  #fiber and pufa unrelated

#t-tests with delivery mode, seq bacth, sex
t.test(ctq_total_score_25it~deliv_mode, data=alpha_diversity_stress_usable) #ns
t.test(ctq_total_score_25it~Sequencing_batch, data=alpha_diversity_stress_usable) #ns
t.test(ctq_total_score_25it~sex_binary, data=alpha_diversity_stress_usable) #ns

t.test(STAI_prorated_state_pw26~sex_binary, data=alpha_diversity_stress_usable) #ns
t.test(STAI_prorated_state_pw26~deliv_mode, data=alpha_diversity_stress_usable) #ns
t.test(STAI_prorated_state_pw26~Sequencing_batch, data=alpha_diversity_stress_usable) #ns

#test whether beta div additional covs are associated with adversity predictors (categorical vars)
#chisquares:deliv mode, seq batch, antibio, ethnciity, bf
chisq.test(alpha_diversity_stress_usable$deliv_mode, alpha_diversity_stress_usable$maltreated_mod_M54) #ns
chisq.test(alpha_diversity_stress_usable$Sequencing_batch, alpha_diversity_stress_usable$maltreated_mod_M54) #ns
chisq.test(alpha_diversity_stress_usable$antibio_M24, alpha_diversity_stress_usable$maltreated_mod_M54) #ns
chisq.test(alpha_diversity_stress_usable$mother_ethnicity, alpha_diversity_stress_usable$maltreated_mod_M54) #ns
chisq.test(alpha_diversity_stress_usable$only_bf_months, alpha_diversity_stress_usable$maltreated_mod_M54) #ns

#t-tests: fiber, pufa
t.test(Fibre.per.1000kcal~maltreated_mod_M54, data=alpha_diversity_stress_usable) #ns
t.test(PUFA_.~maltreated_mod_M54, data=alpha_diversity_stress_usable) #ns

#basic question: are covariates related to predictors? 
#postnatal adversity DA analyses: child ethnicity, other adversities (unrelated)
chisq.test(alpha_diversity_stress_usable$postnatal_stress, alpha_diversity_stress_usable$mother_ethnicity) #ns

chisq.test(alpha_diversity_stress_usable$postnatal_stress, alpha_diversity_stress_usable$only_bf_months) #ns

chisq.test(alpha_diversity_stress_usable$postnatal_stress, alpha_diversity_stress_usable$Sequencing_batch)

t.test(c_age_years_stoolproxy_m24~postnatal_stress, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE)

#prenatal adversity DA analyses: ethnicity, duration of exclusive breastfeeding, sequencing batch, probiotic use
summary(aov(STAI_prorated_state_pw26~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(STAI_prorated_state_pw26~only_bf_months, data=alpha_diversity_stress_usable)) #sig, p = .01
t.test(STAI_prorated_state_pw26~Sequencing_batch, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(STAI_prorated_state_pw26~probiotics_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns

#probe difference in prenatal adversity by duration of exclusive bf
prenat_bf <- alpha_diversity_stress_usable %>% group_by(only_bf_months) %>% summarize(prenat_anx_avg = mean(STAI_prorated_state_pw26, na.rm=T)) #get prenatal anxiety average within each breastfeeding group

TukeyHSD(aov(STAI_prorated_state_pw26~only_bf_months, data=alpha_diversity_stress_usable)) #use Tukey HSD method of post-hoc testing because we are interested in comparing all groups to each other
#only sig difference is between 3-6m months and <1 month (lowest and highest groups)

#developmental problems Y2: child sex, prenatal adversity, child ethnicity, duration of exclusive bf, delivery mode
t.test(cbcldevprobtot_m24_std~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
cor.test(alpha_diversity_stress_usable$cbcldevprobtot_m24_std, alpha_diversity_stress_usable$STAI_prorated_state_pw26, method="pearson") #positive correlation (r=.236, p = .002)
summary(aov(cbcldevprobtot_m24_std~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbcldevprobtot_m24_std~only_bf_months, data=alpha_diversity_stress_usable)) #ns
t.test(cbcldevprobtot_m24_std~deliv_mode, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns

#internalizing Y4: child sex, prenatal adversity, child ethnicity, duration of exclusive bf, sequencing batch
t.test(cbclintprobtot_y4_std~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
cor.test(alpha_diversity_stress_usable$cbclintprobtot_y4_std, alpha_diversity_stress_usable$STAI_prorated_state_pw26, method="pearson") #positive correlation (r=.298, p < .001)
summary(aov(cbclintprobtot_y4_std~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbclintprobtot_y4_std~only_bf_months, data=alpha_diversity_stress_usable)) #ns
t.test(cbclintprobtot_y4_std~Sequencing_batch, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #higher in batch 1

#sleep Y4: child sex, prenatal adversity, child ethnicity, duration of exclusive bf, sequencing batch, probiotic use
t.test(cbclsleepprobtot_y4_std~sex_binary, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
cor.test(alpha_diversity_stress_usable$cbclsleepprobtot_y4_std, alpha_diversity_stress_usable$STAI_prorated_state_pw26, method="pearson") #positive correlation (r=.21, p < .001)
summary(aov(cbclsleepprobtot_y4_std~mother_ethnicity, data=alpha_diversity_stress_usable)) #ns
summary(aov(cbclsleepprobtot_y4_std~only_bf_months, data=alpha_diversity_stress_usable)) #ns
t.test(cbclsleepprobtot_y4_std~Sequencing_batch, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
t.test(cbclsleepprobtot_y4_std~probiotics_M24, data=alpha_diversity_stress_usable, alternative="two.sided", paired=FALSE) #ns
```

### Differential abundance analyses at family and genus level
Here we will only test prenatal adversity (only adversity in model), developmental problems at Y2, and sleep problems at Y4 because we found taxa from the same family to be DA in those domains and these analyses are specifically to follow-up on those results. 
#### Format family-level data for maaslin2
2. A tab-delimited feature file (taxonomy as columns, samples as rows)  
```{r}
M24_meta_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/M24_metadata.tsv"

family_taxonomy <- as.data.frame(read.csv(family_taxonomy_file))
rownames(family_taxonomy) <- family_taxonomy[,1]
family_taxonomy <- family_taxonomy[,2:ncol(family_taxonomy)]
family_taxonomy_filtered <- family_taxonomy %>% 
filter(!(if_all(everything(.), ~. == 0))) #filter taxa for which all samples have zero abundance 
family_taxonomy_filtered <- as.data.frame(t(family_taxonomy_filtered))
family_taxonomy_filtered <- family_taxonomy_filtered %>% 
  rename('#SampleID' = '#OTU ID') 
write_tsv(family_taxonomy_filtered, "../../data/analysis_datasets/fran/maaslin2/inputs/family_tax_features.tsv")

family_tax_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/family_tax_features.tsv"
```

#### Format genus-level data for maaslin2
```{r}
genus_taxonomy <- as.data.frame(read.csv(genus_taxonomy_file))
rownames(genus_taxonomy) <- genus_taxonomy[,1]
genus_taxonomy <- genus_taxonomy[,2:ncol(genus_taxonomy)]
genus_taxonomy_filtered <- genus_taxonomy %>% 
  filter(!(if_all(everything(.), ~. == 0))) #filter taxa for which all samples have zero abundance 
genus_taxonomy_filtered <- as.data.frame(t(genus_taxonomy_filtered))
genus_taxonomy_filtered <- genus_taxonomy_filtered %>% 
  rename('#SampleID' = '#OTU ID') 
write_tsv(genus_taxonomy_filtered, "../../data/analysis_datasets/fran/maaslin2/inputs/genus_tax_features.tsv")

genus_tax_fp <- "../../data/analysis_datasets/fran/maaslin2/inputs/genus_tax_features.tsv"
```


#### Family-level DA for prenatal adversity (only adversity in model)
```{r}
#prenatal adversity -- 0.5% filtering threshold 
prenatanx_covs_0.5p_family <- Maaslin2(
  input_data = family_tax_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/prenatal_anxiety/prenatanx_wcovs_0.5p_allcases_family",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("mother_ethnicity", "only_bf_months", "probiotics_M24", "STAI_prorated_state_pw26", "Sequencing_batch"),
  standardize = TRUE, 
  reference = c('mother_ethnicity, chinese', 'only_bf_months, <1M') 
)
```

#### Genus-level DA for prenatal adversity (only adversity in model)
```{r}
#prenatal adversity -- 0.5% filtering threshold 
prenatanx_covs_0.5p_genus <- Maaslin2(
  input_data = genus_tax_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/prenatal_anxiety/prenatanx_wcovs_0.5p_allcases_genus",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("mother_ethnicity", "only_bf_months", "probiotics_M24", "STAI_prorated_state_pw26", "Sequencing_batch"),
  standardize = TRUE, 
  reference = c('mother_ethnicity, chinese', 'only_bf_months, <1M') 
)
```

#### Family-level DA for developmental problems at Y2
```{r}
#NOTE: this is the version with ethnicity added

maaslin_dev2_ethadd_0.5p_family = Maaslin2(
  input_data = family_tax_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/dev_raw_0.5p_results_allcases_ethadded_family",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcldevprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch", "deliv_mode_str", "only_bf_months", "Sequencing_batch", "mother_ethnicity"),
  standardize = TRUE,
  reference = 'mother_ethnicity, chinese' 
) 
```


#### Genus-level DA for developmental problems at Y2
```{r}
#NOTE: this is the version with ethnicity added

maaslin_dev2_ethadd_0.5p_family = Maaslin2(
  input_data = genus_tax_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/m24/with_covs/dev_raw_0.5p_results_allcases_ethadded_genus",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", #default
  fixed_effects = c("cbcldevprobtot_m24_std", "STAI_prorated_state_pw26", "sex_binary_ch", "deliv_mode_str", "only_bf_months", "Sequencing_batch", "mother_ethnicity"),
  standardize = TRUE,
  reference = 'mother_ethnicity, chinese' 
) 
```

#### Family-level DA for sleep problems at Y4
```{r}
maaslin_sleep4_0.5p_family = Maaslin2(
  input_data = family_tax_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/cbcl/y4/with_covs/sleep_raw_0.5p_results_allcases_family",
  min_abundance = 0,
  min_prevalence = 0.005,
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("STAI_prorated_state_pw26", "cbclsleepprobtot_y4_std", "sex_binary_ch", "only_bf_months", "mother_ethnicity", "probiotics_M24", "Sequencing_batch"),
  standardize = TRUE, 
  reference = 'mother_ethnicity, c(0, 1, 2)'
)
```


## CTQ subscales and gut microbiome associations
Were some maltreatment subtypes more likely to be associated with offspring microbiomes?
### Alpha diversity
Within model controlling for prenatal and postnatal adversities, same covariats as for other alpha div analyses
```{r}
alpha_div_all_cov_complete <- #create complete cases dataset for this analysis
  alpha_diversity_stress_usable %>% 
  filter(!is.na(ctq_total_score_25it) & !is.na(STAI_prorated_state_pw26) & !is.na(postnatal_stress) & !is.na(deliv_mode) & !is.na(sex_binary) &!is.na(monthly_income_per_member) &!is.na(Sequencing_batch) & !is.na(PUFA_.)) 

#define models (pn, en, ea, sa, pa subscales)
shannon_pn_covs <- lm(shannon_entropy~ctq_pn_score_3it+STAI_prorated_state_pw26+postnatal_stress+sex_binary+deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_div_all_cov_complete)
shannon_pn_ncovs <- lm(shannon_entropy~ctq_pn_score_3it+STAI_prorated_state_pw26+postnatal_stress, data=alpha_div_all_cov_complete)

shannon_ea_covs <- lm(shannon_entropy~ctq_ea_score+STAI_prorated_state_pw26+postnatal_stress+sex_binary+deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_div_all_cov_complete)
shannon_ea_ncovs <- lm(shannon_entropy~ctq_ea_score+STAI_prorated_state_pw26+postnatal_stress, data=alpha_div_all_cov_complete)

shannon_pa_covs <- lm(shannon_entropy~ctq_pa_score+STAI_prorated_state_pw26+postnatal_stress+sex_binary+deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_div_all_cov_complete)
shannon_pa_ncovs <- lm(shannon_entropy~ctq_pa_score+STAI_prorated_state_pw26+postnatal_stress, data=alpha_div_all_cov_complete)

shannon_sa_covs <- lm(shannon_entropy~ctq_sa_score+STAI_prorated_state_pw26+postnatal_stress+sex_binary+deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_div_all_cov_complete)
shannon_sa_ncovs <- lm(shannon_entropy~ctq_sa_score+STAI_prorated_state_pw26+postnatal_stress, data=alpha_div_all_cov_complete)

shannon_en_covs <- lm(shannon_entropy~ctq_en_score+STAI_prorated_state_pw26+postnatal_stress+sex_binary+deliv_mode+Fibre.per.1000kcal+monthly_income_per_member+PUFA_.+Sequencing_batch, data = alpha_div_all_cov_complete)
shannon_en_ncovs <- lm(shannon_entropy~ctq_en_score+STAI_prorated_state_pw26+postnatal_stress, data=alpha_div_all_cov_complete)

#run models
knitr::knit_print(tab_model(shannon_pn_covs, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE)) #ns

knitr::knit_print(tab_model(shannon_en_covs, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE)) #ns

knitr::knit_print(tab_model(shannon_ea_covs, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE)) #ns

knitr::knit_print(tab_model(shannon_sa_covs, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE)) #ns

knitr::knit_print(tab_model(shannon_pa_covs, robust=TRUE, show.se=TRUE, show.std=TRUE, show.stat=TRUE)) #ns
```

### Differential abundance
Start with 0.5% filtering threshold, preconception adveristy by itself because we didn't see any sig. differences for precon with all adversities entered into same model. 
```{r}
#pn
maaslin_pn_covs_0.5p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/ctq_subscales/pn_wcovs",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "ctq_pn_score_3it", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity,chinese', 'only_bf_months,<1M') 
)

#en
maaslin_en_covs_0.5p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/ctq_subscales/en_wcovs",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "ctq_en_score", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity,chinese', 'only_bf_months,<1M') 
)

#ea
maaslin_ea_covs_0.5p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/ctq_subscales/ea_wcovs",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "ctq_ea_score", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity,chinese', 'only_bf_months,<1M') 
)

#pa
maaslin_pa_covs_0.5p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/ctq_subscales/pa_wcovs",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "ctq_pa_score", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity,chinese', 'only_bf_months,<1M') 
)

#sa
maaslin_sa_covs_0.5p <- Maaslin2(
  input_data = tax_features_ra_fp,
  input_metadata = M24_meta_fp,
  output = "../../data/analysis_datasets/fran/maaslin2/final_results/adversity/with_covs/ctq_subscales/sa_wcovs",
  min_abundance = 0,
  min_prevalence = 0.005, 
  max_significance = 0.25, 
  analysis_method = "lm", 
  fixed_effects = c("mother_ethnicity", "only_bf_months", "ctq_sa_score", "Sequencing_batch", "Fibre.per.1000kcal"),
  standardize = TRUE, 
  reference = c('mother_ethnicity,chinese', 'only_bf_months,<1M') 
)

```

